<div class="container">

<table style="width: 100%;"><tr>
<td>geneDrop</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Functions to conduct gene dropping through a pedigree</h2>

<h3>Description</h3>

<p>Functions that perform and summarize gene dropping conducted on supplied pedigrees
</p>


<h3>Usage</h3>

<pre><code class="language-R">geneDrop(
  pedigree,
  N,
  parallel = FALSE,
  ncores = getOption("mc.cores", 2L),
  ...
)

## Default S3 method:
geneDrop(
  pedigree,
  N,
  parallel = FALSE,
  ncores = getOption("mc.cores", 2L),
  ...
)

## S3 method for class 'numPed'
geneDrop(
  pedigree,
  N,
  parallel = FALSE,
  ncores = getOption("mc.cores", 2L),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>pedigree</code></td>
<td>
<p>A pedigree with columns organized: ID, Dam, Sire.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>N</code></td>
<td>
<p>The number of times to iteratively trace alleles through the
pedigree</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>parallel</code></td>
<td>
<p>A logical indicating whether or not to use parallel
processing. Note, this may only be available for Mac and Linux operating
systems.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncores</code></td>
<td>
<p>The number of cpus to use when constructing the dominance
relatedness matrix. Default is all available.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Other arguments that can be supplied to alter what summaries are
reported.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Missing parents (e.g., base population) should be denoted by either 'NA', '0'
, or '*'.
</p>
<p><code>parallel</code> = TRUE should only be used on Linux or Mac operating systems
(i.e., not Windows).
</p>
<p>Founder allelic values (the alleles assigned to an individual's maternal,
paternal, or both haplotypes when the maternal, paternal, or both parents are
missing) are equivalent positive and negative integer values corresponding to
the maternal and paternal haplotypes, respectively. For example, if the first
individual in the pedigree has two unknown parents it will have the following
two allelic values: 1=maternal haplotype and -1=paternal haplotype.
</p>


<h3>Value</h3>

<p>a <code>list</code>:
</p>

<dl>
<dt>IDs </dt>
<dd>
<p>Original identities in the pedigree</p>
</dd>
<dt>maternal </dt>
<dd>
<p>Simulated maternal haplotypes</p>
</dd>
<dt>paternal </dt>
<dd>
<p>Simulated paternal haplotypes</p>
</dd> 
<dt>numericPedigree </dt>
<dd>
<p>Pedigree in class <code>numPed</code> for convenient
post-processing of haplotypes</p>
</dd>
</dl>
<h3>Author(s)</h3>

<p><a href="mailto:matthewwolak@gmail.com">matthewwolak@gmail.com</a>
</p>


<h3>See Also</h3>

<p><code>makeDsim</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">  geneDrop(Mrode2, N = 10)

</code></pre>


</div>