<div class="container">

<table style="width: 100%;"><tr>
<td>needles</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Needleman-Wunsch simple global alignment algorithm
</h2>

<h3>Description</h3>

<p>The Needleman-Wunsch simple gap algorithm was one of the first methods
introduced for global alignment of biological sequences. The same algorithm
can be used to match cell line names or sample names from two related data
sets; we provide examples in the documentation, using data that accompanies
this package.
</p>
<p>While the <span class="pkg">NameNeedle</span> package can be used for biological sequence
alignment, the <span class="pkg">Biostrings</span> package from Bioconductor contains much more
sophisticated tools for that problem.
</p>


<h3>Usage</h3>

<pre><code class="language-R">needles(pattern, subject, params=defaultNeedleParams)
needleScores(pattern, subjects, params=defaultNeedleParams)
defaultNeedleParams
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>pattern</code></td>
<td>
<p>character string to be matched</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>subject</code></td>
<td>
<p>character string to be matched against</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>subjects</code></td>
<td>
<p>character vector where matches are sought</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>params</code></td>
<td>

<p>list containing four required components. The default values are
specified by the object <code>defaultNeedleParams</code>, which contains
the following values: </p>
<pre>   
    $ MATCH   : num 1
    $ MISMATCH: num -1
    $ GAP     : num -1
    $ GAPCHAR : chr "*"
</pre>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The Needleman-Wunsch global alignment algorithm was one of the first
algorithms used to align <abbr><span class="acronym">DNA</span></abbr>, <abbr><span class="acronym">RNA</span></abbr>, or protein sequences.
The basic algorithm uses dynamic programming to find an optimal alignment
between two sequences, with parameters that specify penalties for
mismatches and gaps and a reward for exact matches. More elaborate
algorithms (not implemented here) make use of matrices with different
penalties depending on different kinds of mismatches. The version
implemented here is based on the Perl implementation in the first
section of Chapter 3 of the book <cite>BLAST</cite>. 
</p>


<h3>Value</h3>

<p>The <code>needles</code> function returns a list with five components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>score</code></td>
<td>
<p>The raw alignment score.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>align1</code></td>
<td>
<p>The final (optimal) alignment for the <code>pattern</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>align2</code></td>
<td>
<p>The final (optimal) alignment for the <code>subject</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sm</code></td>
<td>
<p>The score matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dm</code></td>
<td>
<p>The backtrace matrix.</p>
</td>
</tr>
</table>
<p>The <code>needleScores</code> function returns a numeric vector the same
length as the <code>subjects</code> argument, with each entry equal to the
corresponding raw alignment score.
</p>


<h3>Author(s)</h3>

<p>Kevin R. Coombes <a href="mailto:krc@silicovore.com">krc@silicovore.com</a>,
P. Roebuck <a href="mailto:proebuck@mdanderson.org">proebuck@mdanderson.org</a>
</p>


<h3>References</h3>

<p>Needleman SB, Wunsch CD.<br><em>A general method applicable to the search for similarities in the
amino acid sequence of two proteins.</em><br>
J Mol Biol 1970, 48(3):443–453.
</p>
<p>Korf I, Yandell M, Bedell J.<br><em>BLAST.</em><br>
O'Reilly Media, 2003.
</p>
<p>Wang J, Byers LA, Yordy JS, Liu W, Shen L, Baggerly KA, Giri U, Myers
JN, Ang KK, Story MD, Girard L, Minna JD, Heymach JV, Coombes KR.<br><em>Blasted cell line names.</em><br>
Cancer Inform. 2010; 9:251–5.
</p>


<h3>See Also</h3>

<p>The <span class="pkg">Biostrings</span> package from Bioconductor used to contain a function
called <code>needwunQS</code> that provided a simple gap implementation of
Needleman-Wunsch, similar to the one presented here. That function has been
deprecated in favor of a more elaborate interface called
<code>pairwiseAlignment</code> that incorporates a variety of
other alignment methods in addition. While <code>pairwiseAlignment</code> is much
more useful for applications to biological sequences, it is serious overkill
for the application we have in mind for matching cell line or other sample
names.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(cellLineNames)
myParam &lt;- defaultNeedleParams
myParam$MATCH &lt;- 2
myParam$MISMATCH &lt;- -2
needles(sf2Names[2], illuNames[1], myParam)
scores &lt;- needleScores(sf2Names[6], illuNames, myParam)
w &lt;- which(scores == max(scores))
w
sf2Names[6]

needles(sf2Names[6], illuNames[w], myParam)
</code></pre>


</div>