<div class="container">

<table style="width: 100%;"><tr>
<td>trainNBLDA</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Train Model over Different Tuning Parameters</h2>

<h3>Description</h3>

<p>This function fits the Negative Binomial classifier using various model parameters and finds the best model parameter using the resampling based performance measures.
</p>


<h3>Usage</h3>

<pre><code class="language-R">trainNBLDA(x, y, type = c("mle", "deseq", "quantile", "tmm"),
  tuneLength = 10, metric = c("accuracy", "error"), train.control = nbldaControl(), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>a n-by-p data frame or matrix. Samples should be in the rows and variables in the columns. Used to train the classifier.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>y</code></td>
<td>
<p>a vector of length n. Each element corresponds to a class label of a sample. Integer and/or factor types are allowed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p>a character string indicating the type of normalization method within the NBLDA model. See details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tuneLength</code></td>
<td>
<p>a positive integer. This is the total number of levels to be used while tuning the model parameter(s) in grid search.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>metric</code></td>
<td>
<p>which criteria should be used while determining the best model parameter? overall accuracy or average number of misclassified samples?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>train.control</code></td>
<td>
<p>a list with control parameters to be used in the NBLDA model. See nbldaControl for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>further arguments. Deprecated.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>NBLDA is proposed to classify count data from any field, e.g., economics, social sciences, genomics, etc. In RNA-Seq studies, for example, normalization is used to adjust between-sample differences for downstream analysis. <code>type</code> is used to define normalization method. Available options are "mle", "deseq", "quantile", and "tmm". Since "deseq", "quantile", and "tmm" methods are originally proposed as robust methods to be used in RNA-Sequencing studies, one should carefully define normalization types. In greater detail, "deseq" estimates the size factors by dividing each sample by the geometric means of the transcript counts (Anders and Huber, 2010). "tmm" trims the lower and upper side of the data by log-fold changes to minimize the log-fold changes between the samples and by absolute intensity (Robinson and Oshlack, 2010). "quantile" is quantile normalization approach of Bullard et al. (2010). "mle" (less robust) divides total counts of each sample to the total counts (Witten, 2010). See related papers for mathematical backgrounds.
</p>


<h3>Value</h3>

<p>an <code>nblda</code> object with following slots:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>input</code></td>
<td>
<p>an <code>nblda_input</code> object including the raw count data and response variable. See <code>nblda_input</code> for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>result</code></td>
<td>
<p>an <code>nblda_trained</code> object including the results from cross-validated and final models. See <code>nblda_trained</code> for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>call</code></td>
<td>
<p>a call expression.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Dincer Goksuluk
</p>


<h3>References</h3>

<p>Witten, DM (2011). Classification and clustering of sequencing data using a Poisson model. Ann. Appl. Stat. 5(4), 2493â€“2518. doi:10.1214/11-AOAS493.
</p>
<p>Dong, K., Zhao, H., Tong, T., &amp; Wan, X. (2016). NBLDA: negative binomial linear discriminant analysis for RNA-Seq data. BMC Bioinformatics, 17(1), 369. http://doi.org/10.1186/s12859-016-1208-1.
</p>
<p>Anders S. Huber W. (2010). Differential expression analysis for sequence count data. Genome Biology, 11:R106
</p>
<p>Witten D. et al. (2010) Ultra-high throughput sequencing-based small RNA discovery and discrete statistical biomarker analysis in a collection of cervical tumours and matched controls. BMC Biology, 8:58
</p>
<p>Robinson MD, Oshlack A (2010). A scaling normalization method for differential expression analysis of RNA-Seq data. Genome Biology, 11:R25, doi:10.1186/gb-2010-11-3-r25
</p>


<h3>Examples</h3>

<pre><code class="language-R">set.seed(2128)
counts &lt;- generateCountData(n = 20, p = 10, K = 2, param = 1, sdsignal = 0.5, DE = 0.8,
                            allZero.rm = FALSE, tag.samples = TRUE)
x &lt;- t(counts$x + 1)
y &lt;- counts$y
xte &lt;- t(counts$xte + 1)
ctrl &lt;- nbldaControl(folds = 2, repeats = 2)

fit &lt;- trainNBLDA(x = x, y = y, type = "mle", tuneLength = 10,
                  metric = "accuracy", train.control = ctrl)

fit
nbldaTrained(fit)  # Cross-validated model summary.

</code></pre>


</div>