<div class="container">

<table style="width: 100%;"><tr>
<td>nbldaControl</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Control parameters for trained NBLDA model.</h2>

<h3>Description</h3>

<p>Define control parameters to be used within <code>trainNBLDA</code> function.
</p>


<h3>Usage</h3>

<pre><code class="language-R">nbldaControl(folds = 5, repeats = 2, foldIdx = NULL, rhos = NULL,
  beta = 1, prior = NULL, transform = FALSE, alpha = NULL, truephi = NULL,
  target = 0, phi.epsilon = 0.15, normalize.target = FALSE, delta = NULL,
  multicore = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>folds</code></td>
<td>
<p>A positive integer. The number of folds for k-fold model validation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>repeats</code></td>
<td>
<p>A positive integer. This is the number of repeats for k-fold model validation. If NULL, 0 or negative, it is set to 1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>foldIdx</code></td>
<td>
<p>a list with indices of hold-out samples for each fold. It should be a list where folds are nested within repeats. If NULL, <code>folds</code> and <code>repeats</code> are used to define hold-out samples.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rhos</code></td>
<td>
<p>A vector of tuning parameters that control the amount of soft thresholding performed. If NULL, it is automatically generated within <code>trainNBLDA</code> using <code>tuneLength</code>, i.e., the length of grid search. See details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>beta</code></td>
<td>
<p>A smoothing term. A Gamma(beta,beta) prior is used to fit the Poisson model. Recommendation is to just leave it at 1, the default value. See Witten (2011) and Dong et al. (2016) for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prior</code></td>
<td>
<p>A vector with a length equal to the number of classes indicates the prior class probabilities. If NULL, all classes are assumed to be equally distributed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>transform</code></td>
<td>
<p>a logical. If TRUE, count data is transformed using power transformation. If <code>alpha</code> is not specified the power transformation parameter is automatically calculated using a goodness-of-fit test. See Witten (2011) for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p>a numeric value within [0, 1] to be used for power transformation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>truephi</code></td>
<td>
<p>a vector with a length equal to the number of variables.  Its elements represent the true overdispersion parameters for each variable. If a single value is given, it is recycled for all variables. If a vector whose length is not equal to the number of variables given, the first element of this vector is used and recycled for all variables. If NULL, estimated overdispersions are used in the classifier. See details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>target</code></td>
<td>
<p>a value for the shrinkage target of dispersion estimates. If NULL, then then a value that is small and minimizes the average squared difference is automatically used as the target value. See <code>getT</code> for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>phi.epsilon</code></td>
<td>
<p>a positive value for controlling the number of features whose dispersions are shrinked towards 0. See details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>normalize.target</code></td>
<td>
<p>a logical. If TRUE and <code>target</code> is NULL, the target value is estimated using the normalized dispersion estimates. See <code>getT</code> for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>delta</code></td>
<td>
<p>a weight within the interval [0, 1] that is used while shrinking dispersions towards 0. When "delta = 0", initial dispersion estimates are forced to be shrunk to 1. Similarly, if "delta = 0", no shrinkage is performed on the initial estimates.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>multicore</code></td>
<td>
<p>a logical. If a parallel backend is loaded and available, the function runs in parallel setting for speeding up the computations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>further arguments passed to <code>trainNBLDA</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><code>rhos</code> is used to control the level of sparsity, i.e., the number of variables (or features) used in the classifier. If a variable has no contribution to the discrimination function, it should be removed from the model. By setting rhos within the interval [0, Inf], it is possible to control the number of variables that are removed from the model. As the upper bound of rhos decreases towards 0, fewer variables are removed. If <code>rhos = 0</code>, all variables are included in the classifier.
</p>
<p><code>truephi</code> controls how the Poisson model differs from the Negative Binomial model. If overdispersion is zero, the Negative Binomial model converges to the Poisson model. Hence, the results from <code>trainNBLDA</code> are identical to PLDA results from <code>Classify</code> when truephi = 0.
</p>
<p><code>phi.epsilon</code> is a value used to shrink estimated overdispersions towards 0. The Poisson model assumes that there is no overdispersion in the observed counts. However, this is not a valid assumption in highly overdispersed count data. <code>NBLDA</code> performs a shrinkage on estimated overdispersions. Although the amount of shrinkage is dependent on several parameters such as <code>delta</code>, <code>target</code>, and <code>truephi</code>, some of the shrunken overdispersions might be very close to 0. By defining a threshold value for shrunken overdispersions, it is possible to shrink very small overdispersions towards 0. If estimated overdispersion is below <code>phi.epsilon</code>, it is shrunken to 0. If <code>phi.epsilon</code> = NULL, threshold value is set to 0. Hence, all the variables with very small overdispersion are included in the NBLDA model.
</p>


<h3>Value</h3>

<p>a list with all the control elements.
</p>


<h3>Author(s)</h3>

<p>Dincer Goksuluk
</p>


<h3>References</h3>

<p>Witten, DM (2011). Classification and clustering of sequencing data using a Poisson model.
Ann. Appl. Stat. 5(4), 2493â€“2518. doi:10.1214/11-AOAS493.
</p>
<p>Dong, K., Zhao, H., Tong, T., &amp; Wan, X. (2016). NBLDA: negative binomial linear discriminant analysis for RNA-Seq data.
BMC Bioinformatics, 17(1), 369. http://doi.org/10.1186/s12859-016-1208-1.
</p>
<p>Yu, D., Huber, W., &amp; Vitek, O. (2013). Shrinkage estimation of dispersion in Negative Binomial models
for RNA-seq experiments with small sample size. Bioinformatics, 29(10), 1275-1282.
</p>


<h3>See Also</h3>

<p><code>getT</code>, <code>getAdjustDisp</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">nbldaControl()  # return default control parameters.

</code></pre>


</div>