<div class="container">

<table style="width: 100%;"><tr>
<td>arab</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Arabidopsis RNA-Seq Data Set</h2>

<h3>Description</h3>

<p>An RNA-Seq dataset from a pilot study of the defense
response of Arabidopsis to infection by bacteria. We
performed RNA-Seq experiments on three independent
biological samples from each of the two treatment groups.
The matrix contains the frequencies of RNA-Seq reads mapped
to genes in a reference database. Rows correspond to genes
and columns correspond to independent biological samples.
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(arab)
</code></pre>


<h3>Format</h3>

<p>A 26222 by 6 matrix of RNA-Seq read frequencies.</p>


<h3>Details</h3>

<p>We challenged leaves of Arabidopsis with the
defense-eliciting <em><code class="reqn">\Delta</code>hrcC</em> mutant of
<em>Pseudomonas syringae</em> pathovar <em>tomato</em> DC3000.
We also infiltrated leaves of Arabidopsis with 10mM MgCl2
as a mock inoculation. RNA was isolated 7 hours after
inoculation, enriched for mRNA and prepared for RNA-Seq. We
sequenced one replicate per channel on the Illumina Genome
Analyzer (http://www.illumina.com).  The length of the
RNA-Seq reads can vary in length depending on user
preference and the sequencing instrument. The dataset used
here are derived from a 36-cycle sequencing reaction, that
we trimmed to 25mers. We used an in-house computational
pipeline to process, align, and assign RNA-Seq reads to
genes according to a reference database we developed for
Arabidopsis.
</p>


<h3>Author(s)</h3>

<p>Jason S Cumbie <a href="mailto:cumbiej@onid.orst.edu">cumbiej@onid.orst.edu</a> and Jeff H
Chang <a href="mailto:changj@cgrb.oregonstate.edu">changj@cgrb.oregonstate.edu</a>.
</p>


<h3>References</h3>

<p>Di Y, Schafer DW, Cumbie JS, and Chang JH (2011): "The NBP
Negative Binomial Model for Assessing Differential Gene
Expression from RNA-Seq", Statistical Applications in
Genetics and Molecular Biology, 10 (1).
</p>


</div>