<div class="container">

<table style="width: 100%;"><tr>
<td>exact.nb.test</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Exact Negative Binomial Test for Differential Gene Expression</h2>

<h3>Description</h3>

<p><code>exact.nb.test</code> performs the Robinson and Smyth exact
negative binomial (NB) test for differential gene
expression on each gene and summarizes the results using
p-values and q-values (FDR).
</p>


<h3>Usage</h3>

<pre><code class="language-R">exact.nb.test(obj, grp1, grp2, print.level = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>obj</code></td>
<td>
<p>output from <code>estimate.disp</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>grp1</code></td>
<td>
<p>Identifier of group 1. A number, character or
string (should match at least one of the obj$grp.ids).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>grp2</code></td>
<td>
<p>Identifier of group 2. A number, character or
string (should match at least one of the obj$grp.ids).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>print.level</code></td>
<td>
<p>a number.  Controls the amount of
messages printed: 0 for suppressing all messages, 1 for
basic progress messages, larger values for more detailed
messages.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The negative binomial (NB) distribution offers a more
realistic model for RNA-Seq count variability and still
permits an exact (non-asymptotic) test for comparing
expression levels in two groups.
</p>
<p>For each gene, let <code class="reqn">S_1</code>, <code class="reqn">S_2</code> be the sums of gene
counts from all biological replicates in each group. The
exact NB test is based on the conditional distribution of
<code class="reqn">S_1|S_1+S_2</code>: a value of <code class="reqn">S_1</code> that is too big or
too small, relative to the sum <code class="reqn">S_1+S_2</code>, indicates
evidence for differential gene expression.  When the
effective library sizes are the same in all replicates and
the dispersion parameters are known, we can determine the
probability functions of <code class="reqn">S_1</code>, <code class="reqn">S_2</code> explicitly.
The exact p-value is computed as the total conditional
probability of all possible values of <code class="reqn">(S_1, S_2)</code> that
have the same sum as but are more extreme than the observed
values of <code class="reqn">(S_1, S_2)</code>.
</p>
<p>Note that we assume that the NB dispersion parameters for
the two groups are the same and library sizes (column
totals of the count matrix) are the same.
</p>


<h3>Value</h3>

<p>the list <code>obj</code> from the input with the following added
components: </p>
<table>
<tr style="vertical-align: top;">
<td><code>grp1</code></td>
<td>
<p>same as input.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>grp2</code></td>
<td>
<p>same as
input.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pooled.pie</code></td>
<td>
<p>estimated pooled mean of relative
count frequencies in the two groups being compared.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>expression.levels</code></td>
<td>
<p>a matrix of estimated gene
expression levels as indicated by mean relative read
frequencies. It has three columns <code>grp1</code>, <code>grp2</code>,
<code>pooled</code> corresponding to the two treatment groups and
the pooled mean.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>log.fc</code></td>
<td>
<p>base 2 log fold change in
mean relative frequency between two groups.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.values</code></td>
<td>
<p>p-values of the exact NB test applied to
each gene (row).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>q.values</code></td>
<td>
<p>q-values (estimated FDR)
corresponding to the p-values.</p>
</td>
</tr>
</table>
<h3>Note</h3>

<p>Before calling <code>exact.nb.test</code>, the user should
call <code>estimate.norm.factors</code> to estimate
normalization factors, call <code>prepare.nbp</code> to
adjust library sizes, and call <code>estimate.disp</code>
to fit a dispersion model. The exact NB test will be
performed using <code>pseudo.counts</code> in the list
<code>obj</code>, which are normalized and adjusted to have the
same effective library sizes (column sums of the count
matrix, multiplied by normalization factors).
</p>
<p>Users not interested in fine tuning the underlying
statistical model should use <code>nbp.test</code>
instead. The all-in-one function <code>nbp.test</code>
uses sensible approaches to normalize the counts, estimate
the NBP model parameters and test for differential gene
expression.
</p>
<p>A test will be performed on a row (a gene) only when the
total row count is nonzero, otherwise an NA value will be
assigned to the corresponding p-value and q-value.
</p>


<h3>See Also</h3>

<p><code>nbp.test</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Load Arabidopsis data
data(arab);

## Specify treatment groups
## grp.ids = c(1, 1, 1, 2, 2, 2); # Numbers or strings are both OK
grp.ids = rep(c("mock", "hrcc"), each=3);

## Estimate normalization factors
norm.factors = estimate.norm.factors(arab);
print(norm.factors);

## Prepare an NBP object, adjust the library sizes by thinning the
## counts. For demonstration purpose, only use the first 100 rows of
## the arab data.
set.seed(999);
obj = prepare.nbp(arab[1:100,], grp.ids, lib.size=colSums(arab), norm.factors=norm.factors);
print(obj);

## Fit a dispersion model (NBQ by default)
obj = estimate.disp(obj);
plot(obj);

## Perform exact NB test
## grp1 = 1;
## grp2 = 2;
grp1 = "mock";
grp2 = "hrcc";

obj = exact.nb.test(obj, grp1, grp2);

## Print and plot results
print(obj);
par(mfrow=c(3,2));
plot(obj);
</code></pre>


</div>