<div class="container">

<table style="width: 100%;"><tr>
<td>prepare.nbp</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Prepare the Data Structure for Exact NB test for Two-Group Comparison</h2>

<h3>Description</h3>

<p>Create the NBP data structure, (optionally) normalize the
counts, and thin the counts to make the effective library
sizes equal.
</p>


<h3>Usage</h3>

<pre><code class="language-R">prepare.nbp(counts, grp.ids, lib.sizes = colSums(counts),
  norm.factors = NULL, thinning = TRUE, print.level = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>counts</code></td>
<td>
<p>an <code class="reqn">n</code> by <code class="reqn">r</code> matrix of RNA-Seq
read counts with rows corresponding to genes (exons, gene
isoforms, etc) and columns corresponding to libraries
(independent biological samples).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>grp.ids</code></td>
<td>
<p>an <code class="reqn">r</code> vector of treatment group
identifiers (can be a vector of integers, chars or
strings).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lib.sizes</code></td>
<td>
<p>library sizes, an <code class="reqn">r</code> vector of
numbers. By default, library sizes are estimated by
column sums.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>norm.factors</code></td>
<td>
<p>normalization factors, an <code class="reqn">r</code>
vector of numbers. If <code>NULL</code> (default), no
normalization will be applied.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>thinning</code></td>
<td>
<p>a boolean variable (i.e., logical). If
<code>TRUE</code> (default), the counts will be randomly down
sampled to make effective library sizes approximately
equal.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>print.level</code></td>
<td>
<p>a number, controls the amount of
messages printed: 0 for suppressing all messages, 1
(default) for basic progress messages, and 2 to 5 for
increasingly more detailed messages.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Normalization
</p>
<p>We take gene expression to be indicated by relative
frequency of RNA-Seq reads mapped to a gene, relative to
library sizes (column sums of the count matrix). Since the
relative frequencies sum to 1 in each library (one column
of the count matrix), the increased relative frequencies of
truly over expressed genes in each column must be
accompanied by decreased relative frequencies of other
genes, even when those others do not truly differently
express. Robinson and Oshlack (2010) presented examples
where this problem is noticeable.
</p>
<p>A simple fix is to compute the relative frequencies
relative to effective library sizesâ€”library sizes
multiplied by normalization factors. Many authors (Robinson
and Oshlack (2010), Anders and Huber (2010)) propose to
estimate the normalization factors based on the assumption
that most genes are NOT differentially expressed.
</p>
<p>By default, <code>prepare.nbp</code> does not estimate the
normalization factors, but can incorporate user specified
normalization factors through the argument
<code>norm.factors</code>.
</p>
<p>Library Size Adjustment
</p>
<p>The exact test requires that the effective library sizes
(column sums of the count matrix multiplied by
normalization factors) are approximately equal. By default,
<code>prepare.nbp</code> will thin (downsample) the counts to
make the effective library sizes equal. Thinning may lose
statistical efficiency, but is unlikely to introduce bias.
</p>


<h3>Value</h3>

<p>A list containing the following components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>counts</code></td>
<td>
<p>the count matrix, same as input.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lib.sizes</code></td>
<td>
<p>column sums of the count matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>grp.ids</code></td>
<td>
<p>a vector of identifiers of treatment groups,
same as input.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>eff.lib.sizes</code></td>
<td>
<p>effective library
sizes, lib.sizes multiplied by the normalization factors.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pseudo.counts</code></td>
<td>
<p>count matrix after thinning.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pseduo.lib.sizes</code></td>
<td>
<p>effective library sizes of pseudo
counts, i.e., column sums of the pseudo count matrix
multiplied by the normalization.</p>
</td>
</tr>
</table>
<h3>Note</h3>

<p>Due to thinning (random downsampling of counts), two
identical calls to <code>prepare.nbp</code> may yield slightly
different results. A random number seed can be used to make
the results reproducible.
</p>


<h3>See Also</h3>

<p><code>nbp.test</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## See the example for exact.nb.test
</code></pre>


</div>