<div class="container">

<table style="width: 100%;"><tr>
<td>cutoffs</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Cut-points 
</h2>

<h3>Description</h3>

<p>Calculates cut-points of ordinal variables with respect to the Gaussian copula. 
</p>


<h3>Usage</h3>

<pre><code class="language-R">cutoffs(y)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>y</code></td>
<td>

<p>An (<code class="reqn">n \times p</code>) matrix or a <code>data.frame</code> corresponding to the data matrix (<code class="reqn">n</code> is the sample size and <code class="reqn">p</code> is the number of variables). It also could be an object of class <code>"simgeno"</code>.
</p>
</td>
</tr></table>
<h3>Details</h3>

<p> The relationship between <code class="reqn">j</code>th variable and <code class="reqn">j</code>th latent variable is expressed through this set of cut-points.
</p>


<h3>Value</h3>

<table><tr style="vertical-align: top;">
<td><code>cutoffs</code></td>
<td>
<p> A <code class="reqn">p</code> by <code class="reqn">(k + 1)</code> matrix representing the cut-point values under the Gaussian copula, where <code class="reqn">k</code> defines the number of categories in the dataset.</p>
</td>
</tr></table>
<h3>Author(s)</h3>

 
<p>Pariya Behrouzi and Ernst C. Wit <br>
Maintainer: Pariya Behrouzi &lt;pariya.behrouzi@gmail.com&gt;
</p>


<h3>References</h3>

<p>1. Behrouzi, P., and Wit, E. C. (2019). Detecting epistatic selection with partially observed genotype data by using copula graphical models. Journal of the Royal Statistical Society: Series C (Applied Statistics), 68(1), 141-160. <br>
2. Behrouzi, P., and Wit, E. C. (2018). De novo construction of polyploid linkage maps using discrete graphical models. Bioinformatics. <br>
3. Behrouzi, P., and Wit, E. C. (2017c). netgwas: An R Package for Network-Based Genome-Wide Association Studies. arXiv preprint, arXiv:1710.01236.
</p>


<h3>See Also</h3>

<p><code>lower.upper</code>, <code>simgeno</code> and <code>netgwas-package</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">	D &lt;- simgeno(p = 100, n = 50, k = 3)
	cutoffs(D$data)
</code></pre>


</div>