<div class="container">

<table style="width: 100%;"><tr>
<td>simRIL</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Generate genotype data of RIL</h2>

<h3>Description</h3>

<p>Generating genotype data from a recombinant inbred line (RIL) population.
</p>


<h3>Usage</h3>

<pre><code class="language-R">simRIL( d = 25, n = 200, g = 5, cM = 100, selfing=2 )
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>d</code></td>
<td>
<p>The number of markers per chromosome. The default value is 25.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n</code></td>
<td>
<p>The number of sample size (observations). The default value is 200.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>g</code></td>
<td>
<p>The number of linkage groups (chromosomes). The default value is 5.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cM</code></td>
<td>
<p>The length of each chromosome based on centiMorgan.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>selfing</code></td>
<td>
<p> The number of selfing in RIL population.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>The generated RIL genotype data as an <code>n</code> by <code>(d x g)</code> matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map</code></td>
<td>
<p>The genetic map of the data.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

 
<p>Pariya Behrouzi <br>
Maintainer: Pariya Behrouzi &lt;pariya.behrouzi@gmail.com&gt;
</p>


<h3>See Also</h3>

<p><code>netmap</code>, <code>netsnp</code>, and <code>netgwas-package</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">#genome-like graph structure
ril &lt;- simRIL(g = 5, d = 25, cM = 100, n = 200, selfing = 2)
geno &lt;- ril$data; image(geno, xlab= "individuals", ylab="markers")
map &lt;- ril$map 
</code></pre>


</div>