<div class="container">

<table style="width: 100%;"><tr>
<td>selectnet</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Model selection </h2>

<h3>Description</h3>

<p>Estimate the optimal regularization parameter at EM convergence based on different information criteria . 
</p>


<h3>Usage</h3>

<pre><code class="language-R">selectnet(netgwas.obj, opt.index= NULL, criteria= NULL, ebic.gamma=0.5, 
		   ncores= NULL, verbose= TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>netgwas.obj</code></td>
<td>
<p>An object with S3 class "netgwas"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>opt.index</code></td>
<td>
<p>The program internally determines an optimal graph using <code>opt.index= NULL</code>. Otherwise, to manually choose an optimal graph from the graph path.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>criteria</code></td>
<td>
<p>Model selection criteria. "ebic" and "aic" are available. BIC model selection can be calculated by fixing <code>ebic.gamma = 0</code>. Applicable only if <code>opt.index= NULL</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ebic.gamma</code></td>
<td>
<p>The tuning parameter for ebic. The<code>ebic.gamma = 0</code> results in bic model selection. The default value is 0.5. Applicable only <code>opt.index= NULL</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncores</code></td>
<td>
<p> The number of cores to use for the calculations. Using <code>ncores = "all"</code> automatically detects number of available cores and runs the computations in parallel.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>If <code>verbose = FALSE</code>, printing information is disabled. The default value is <code>TRUE</code>. Applicable only <code>opt.index= NULL</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function computes extended Bayesian information criteria (ebic), Bayesian information criteria, Akaike information criterion (aic) at EM convergence based on observed or joint log-likelihood.
The observed log-likelihood can be obtained through  
</p>
<p style="text-align: center;"><code class="reqn">\ell_Y(\widehat{\Theta}_\lambda) = Q(\widehat{\Theta}_\lambda | \widehat{\Theta}^{(m)}) - H (\widehat{\Theta}_\lambda | \widehat{\Theta}^{(m)}),</code>
</p>

<p>Where <code class="reqn">Q</code> can be calculated from <code>netmap</code>, <code>netsnp</code>, <code>netphenogeno</code> function and H function is 
</p>
<p style="text-align: center;"><code class="reqn">H(\widehat{\Theta}_\lambda | \widehat{\Theta}^{(m)}_\lambda) = E_z[\ell_{Z | Y}(\widehat{\Theta}_\lambda) | Y; \widehat{\Theta}_\lambda] = E_z[\log f(z)| Y ;\widehat{\Theta}_\lambda ] - \log p(y).</code>
</p>

<p>The "ebic" and "aic" model selection criteria can be obtained as follow
</p>
<p style="text-align: center;"><code class="reqn">ebic(\lambda) = -2 \ell(\widehat{\Theta}_\lambda) +  ( \log n + 4 \gamma \log p) df(\lambda)</code>
</p>

<p style="text-align: center;"><code class="reqn">aic(\lambda) = -2 \ell(\widehat{\Theta}_\lambda) +  2 df(\lambda)</code>
</p>

<p>where <code class="reqn">df</code> refers to the number of non-zeros offdiagonal elements of <code class="reqn">\hat{\Theta}_\lambda</code>, and <code class="reqn">\gamma \in [0, 1]</code>. Typical value for for <code>ebic.gamma</code> is 1/2, but it can also be tuned by experience. Fixing <code>ebic.gamma = 0 </code> results in bic model selection.
</p>


<h3>Value</h3>

<p>An obj with S3 class "selectnet" is returned:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>opt.adj</code></td>
<td>
<p> The optimal graph selected from the graph path</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>opt.theta</code></td>
<td>
<p>The optimal precision matrix from the graph path</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>opt.sigma</code></td>
<td>
<p> The optimal covariance matrix from the graph path</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ebic.scores</code></td>
<td>
<p> Extended BIC scores for regularization parameter selection at the EM convergence. Applicable if <code>opt.index = NULL</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>opt.index</code></td>
<td>
<p> The index of optimal regularization parameter. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>opt.rho</code></td>
<td>
<p> The selected regularization parameter. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>par.cor</code></td>
<td>
<p> A partial correlation matrix. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>V.names</code></td>
<td>
<p> Variables name whose are not isolated. </p>
</td>
</tr>
</table>
<p>and anything else that is included in the input <code>netgwas.obj</code>.
</p>


<h3>Author(s)</h3>

 
<p>Pariya Behrouzi and Ernst C.Wit <br>
Maintainer: Pariya Behrouzi <a href="mailto:pariya.behrouzi@gmail.com">pariya.behrouzi@gmail.com</a> <br></p>


<h3>References</h3>

<p>1. BBehrouzi, P., and Wit, E. C. (2019). Detecting epistatic selection with partially observed genotype data by using copula graphical models. Journal of the Royal Statistical Society: Series C (Applied Statistics), 68(1), 141-160. <br>
2. Behrouzi, P., Arends, D., and Wit, E. C. (2023). netgwas: An R Package for Network-Based Genome-Wide Association Studies. The R journal, 14(4), 18-37. <br>
3. Ibrahim, Joseph G., Hongtu Zhu, and Niansheng Tang. (2012). Model selection criteria for missing-data problems using the EM algorithm. Journal of the American Statistical Association.
4. D. Witten and J. Friedman. (2011). New insights and faster computations for the graphical lasso. <em>Journal of Computational and Graphical Statistics</em>, to appear. <br>
5. J. Friedman, T. Hastie and R. Tibshirani. (2007). Sparse inverse covariance estimation with the lasso, <em>Biostatistics</em>.<br>
6. Foygel, R. and M. Drton. (2010). Extended bayesian information criteria for Gaussian graphical models. In Advances in Neural Information Processing Systems, pp. 604-612.
</p>


<h3>See Also</h3>

<p><code>netmap</code>, <code>netsnp</code>, <code>netphenogeno</code></p>


<h3>Examples</h3>

<pre><code class="language-R">	
		   
		#simulate data
		D &lt;- simgeno(p=50, n=100, k= 3, adjacent = 3, alpha = 0.06 , beta = 0.06)
		plot(D)

		#explore intra- and inter-chromosomal interactions
		out  &lt;-  netsnp(D$data, n.rho= 5, ncores= 1)
		plot(out)

		#different graph selection methods
		sel.ebic1 &lt;- selectnet(out, criteria = "ebic")
		plot(sel.ebic1, vis = "CI")

		sel.aic &lt;- selectnet(out, criteria = "aic")
		plot(sel.aic, vis = "CI")

		sel.bic &lt;- selectnet(out, criteria = "ebic", ebic.gamma = 0)
		plot(sel.bic, vis = "CI")
	
</code></pre>


</div>