<div class="container">

<table style="width: 100%;"><tr>
<td>simgeno</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Generate genotype data based on Gaussian copula</h2>

<h3>Description</h3>

<p>Generating discrete ordinal data based on underlying "genome-like" graph structure. The procedure of simulating data relies on a continues variable, which can be simulated from either multivariate normal distribution, or multivariate t-distribution with <code>d</code> degrees of freedom.   
</p>


<h3>Usage</h3>

<pre><code class="language-R">simgeno( p = 90, n = 200, k = NULL, g = NULL, adjacent = NULL, alpha =
              NULL , beta = NULL, con.dist = "Mnorm", d = NULL, vis = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>p</code></td>
<td>
<p>The number of variables. The default value is 90.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n</code></td>
<td>
<p>The number of sample size (observations). The default value is 200.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>k</code></td>
<td>
<p>The number of states (categories). The default value is 3.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>g</code></td>
<td>
<p>The number of groups (chromosomes) in the graph. The default value is about <code class="reqn">p/20</code> if <code class="reqn">p &gt;= 40</code> and 2 if <code class="reqn">p &lt; 40</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>adjacent</code></td>
<td>
<p> The number of adjacent variable(s) to be linked to a variable. For example, if <code>adjacent = 1</code> indicates a variable is linked via an edge with its adjacent variable on the left hand side, and its adjacent variable on the right hand side. The <code>adjacent = 2</code> defines a variable is linked via an edge with its 2 adjacent variables on its left hand side, and 2 adjacent variables on its right hand side. The default value is 1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p> A probability that a pair of non-adjacent variables in the same group is given an edge. The default value is 0.01.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>beta</code></td>
<td>
<p> A probability that variables in different groups are linked with an edge. The default value is 0.02.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>con.dist</code></td>
<td>
<p> The distribution of underlying continuous variable. If <code>con.dist = "Mnorm"</code>, a multivariate normal distribution with mean 0 is applied. If <code>con.dist = "Mt"</code>, the t-distribution with <code>a</code> degrees of freedom is applied. 
The default distribution is <code>con.dist = "Mnorm"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>d</code></td>
<td>
<p>The degrees of freedom of the continuous variable, only applicable when 
<code>con.dist = "Mt"</code>. The default value is 3.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>vis</code></td>
<td>
<p> Visualize the graph pattern and the adjacency matrix of the true graph structure. The default value is TRUE.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The graph pattern is generated as below: <br><br>
genome-like: <code>p</code> variables are evenly partitions variables into <code>g</code> disjoint groups; the adjacent variables within each group are linked via an edge. With a probability <code>alpha</code> a pair of non-adjacent variables in the same group is given an edge. Variables in different groups are linked with an edge with a probability of <code>beta</code>.
</p>


<h3>Value</h3>

<p>An object with S3 class "simgeno" is returned: <br><br></p>
<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>The generated data as an <code>n</code> by <code>p</code> matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Theta</code></td>
<td>
<p>A <code>p</code> by <code>p</code> matrix corresponding to the inverse of covariance.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>adj</code></td>
<td>
<p> A <code>p</code> by <code>p</code> matrix corresponding to the adjacency matrix of the true graph structure.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Sigma</code></td>
<td>
<p> A <code>p</code> by <code>p</code> covariance matrix for the generated data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.groups</code></td>
<td>
<p> The number of groups.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>groups</code></td>
<td>
<p>A vector that indicates each variable belongs to which group.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sparsity</code></td>
<td>
<p>The sparsity levels of the true graph.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

 
<p>Pariya Behrouzi and Ernst C. Wit <br>
Maintainer: Pariya Behrouzi &lt;pariya.behrouzi@gmail.com&gt;
</p>


<h3>References</h3>

<p>1. Behrouzi, P., Arends, D., and Wit, E. C. (2023). netgwas: An R Package for Network-Based Genome-Wide Association Studies. The R journal, 14(4), 18-37.<br>
2. Behrouzi, P., and Wit, E. C. (2019). Detecting epistatic selection with partially observed genotype data by using copula graphical models. Journal of the Royal Statistical Society: Series C (Applied Statistics), 68(1), 141-160. <br></p>


<h3>See Also</h3>

<p><code>netsnp</code>, and <code>netgwas-package</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">#genome-like graph structure
sim1 &lt;- simgeno(alpha = 0.01, beta = 0.02)
plot(sim1)

#genome-like graph structure with more edges between variables in a same or different groups
sim2 &lt;- simgeno(adjacent = 3, alpha = 0.02 , beta = 0.03)
plot(sim2)

		#simulate data
		D &lt;- simgeno(p=50, n=100, g=5, k= 3, adjacent = 3, alpha = 0.06 , beta = 0.08)
		plot(D)

		#Reconstructing intra- and inter-chromosomal conditional interactions (LD) network
		out &lt;- netsnp(data = D$data, n.rho= 4, ncores= 1)
		plot(out)
		#Select an optimal graph
		sel &lt;- selectnet(out)
		plot(sel, vis= "CI" )
	
</code></pre>


</div>