<div class="container">

<table style="width: 100%;"><tr>
<td>buildMap</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> linkage group detection and ordering markers for class "netgwasmap"</h2>

<h3>Description</h3>

<p>Implements different algorithms for detecting linkage groups and ordering markers in each linkage group.
</p>


<h3>Usage</h3>

<pre><code class="language-R">buildMap( res, opt.index, min.m = NULL, use.comu = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>res</code></td>
<td>
<p> An object with S3 class "netgwasmap"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>opt.index</code></td>
<td>
<p> An index of a desired regularization parameter.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.m</code></td>
<td>
<p>Expected minimum number of markers in a chromosome. Optional</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>use.comu</code></td>
<td>
<p>Using community detection algorithm to detect linkage groups. Default is FALSE.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function determines linkage groups and order markers within each linkage group for class "netgwasmap".
</p>


<h3>Value</h3>

<p>An object with S3 class <code>"netgwasmap"</code> is returned:  
</p>
<table>
<tr style="vertical-align: top;">
<td><code>map</code></td>
<td>
<p> Constructed linkage map associated with <code>opt.index</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>opt.index</code></td>
<td>
<p> The index of a desired 3-D map to construct linkage map.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cross</code></td>
<td>
<p>The specified cross type by user. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>allres</code></td>
<td>
<p> A list containing results for different regularization parameter. Belongs to class "netgwas". To visualize a path of different 3D maps consider function <code>plot.netgwas</code>. Note that the input data is reordered based on the estimated linkage map and is saved as data in this argument.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>man</code></td>
<td>
<p>stays TRUE.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

 
<p>Pariya Behrouzi and Ernst C.Wit <br>
Maintainer: Pariya Behrouzi <a href="mailto:pariya.behrouzi@gmail.com">pariya.behrouzi@gmail.com</a> <br></p>


<h3>References</h3>

<p>1. Behrouzi, P., and Wit, E. C. (2018). De novo construction of polyploid linkage maps using discrete graphical models. Bioinformatics. <br>
2. Behrouzi, P., and Wit, E. C. (2017c). netgwas: An R Package for Network-Based Genome-Wide Association Studies. arXiv preprint, arXiv:1710.01236.
</p>


<h3>See Also</h3>

<p><code>netmap</code></p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
data(CviCol)
#Randomly change the order of markers across the genome
cvicol &lt;- CviCol[ ,sample(ncol(CviCol))]

#Constructing linkage map for Cvi x Col genotype data
out &lt;- netmap(cvicol, cross= "inbred", ncores=1); out
plot(out)
map &lt;- out$map; map

#Visualization of other networks
plot(out$allres)  
#Constructing a linkage map for 5th network
bm &lt;- buildMap(out, opt.index=5); bm
plot(bm, vis= "summary")
#or
plot(bm, vis= "interactive", label.vertex="all")

## End(Not run)
</code></pre>


</div>