<div class="container">

<table style="width: 100%;"><tr>
<td>run_netmhc2pan</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Run NetMHCIIpan</h2>

<h3>Description</h3>

<p>Run NetMHCIIpan
</p>


<h3>Usage</h3>

<pre><code class="language-R">run_netmhc2pan(
  fasta_filename,
  alleles = "DRB1_0101",
  peptide_length = 15,
  netmhc2pan_folder_name = get_default_netmhc2pan_folder(),
  temp_xls_filename = netmhc2pan::create_temp_xls_filename()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>fasta_filename</code></td>
<td>
<p>the name of a FASTA file with protein sequences</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alleles</code></td>
<td>
<p>one or more alleles, e.g. <code>DRB1_0101</code>.
See get_netmhc2pan_alleles for a full list</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>peptide_length</code></td>
<td>
<p>length of a peptide</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>netmhc2pan_folder_name</code></td>
<td>
<p>the folder (to be) used by NetMHCIIpan.
From this location, a subfolder for NetMHCIIpan is created.
Use get_default_netmhc2pan_folder to see the location
of the default NetMHCIIpan folder.
Use get_default_netmhc2pan_subfolder to see the location
of the default NetMHCIIpan subfolder.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>temp_xls_filename</code></td>
<td>
<p>name for a temporary <code>xls</code> file,
which will be deleted automatically</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a data frame with the NetMHCIIpan results
</p>


<h3>Author(s)</h3>

<p>Rich√®l J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class="language-R">if (is_netmhc2pan_installed()) {

  fasta_filename &lt;- system.file(
    "extdata", "example.fasta", package = "netmhc2pan"
  )
  run_netmhc2pan(fasta_filename)

  # Two alleles
  alleles &lt;- c("DRB1_0101", "DRB1_0102")
  # Run NetMHCpan with these two alleles
  run_netmhc2pan(fasta_filename, alleles = alleles)
}
</code></pre>


</div>