<div class="container">

<table style="width: 100%;"><tr>
<td>default_params_doc</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>This function does nothing. It is intended to inherit is parameters'
documentation.</h2>

<h3>Description</h3>

<p>This function does nothing. It is intended to inherit is parameters'
documentation.
</p>


<h3>Usage</h3>

<pre><code class="language-R">default_params_doc(
  alleles,
  do_filter,
  fasta_filename,
  folder_name,
  mhc_haplotype,
  netmhc2pan_archive_filename,
  netmhc2pan_bin_tarfile_path,
  netmhc2pan_bin_url,
  netmhc2pan_data_tarfile_path,
  netmhc2pan_data_url,
  netmhc2pan_folder_name,
  netmhc2pan_subfolder,
  netmhc2pan_version,
  os,
  peptide_length,
  peptides,
  protein_sequence,
  temp_fasta_filename,
  temp_local_file,
  temp_xls_filename,
  verbose,
  xls_filename
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>alleles</code></td>
<td>
<p>one or more alleles, e.g. <code>DRB1_0101</code>.
See get_netmhc2pan_alleles for a full list</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>do_filter</code></td>
<td>
<p>set to TRUE if the results of NetMHCIIpan must be filtered</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fasta_filename</code></td>
<td>
<p>the name of a FASTA file with protein sequences</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>folder_name</code></td>
<td>
<p>the folder to install NetMHCIIpan, which is
<code>"/home/[usename]/.local/share"</code> by default</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mhc_haplotype</code></td>
<td>
<p>one MHC haplotype, e.g. <code>DRB1_0101</code>.
See get_netmhc2pan_alleles for a full list</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>netmhc2pan_archive_filename</code></td>
<td>
<p>the NetMHC2pan archive
filename, for example <code>netMHCIIpan-3.2.Linux.tar.gz</code>,
which is used in installation of NetMHC2pan.
Use get_netmhc2pan_archive_filename to get this filename.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>netmhc2pan_bin_tarfile_path</code></td>
<td>
<p>path of the
NetMHCIIpan binary tarball file</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>netmhc2pan_bin_url</code></td>
<td>
<p>URL to download the
NetMHCIIpan binary tarball file from,
similar to, for example,
<code>https://www.cbs.dtu.dk/download/12345678-1234-1234-1234-123456789ABC</code>.
Use get_netmhc2pan_bin_url to get the default URL.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>netmhc2pan_data_tarfile_path</code></td>
<td>
<p>path of the
NetMHCIIpan data tarball file</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>netmhc2pan_data_url</code></td>
<td>
<p>URL to download the
NetMHCIIpan data tarball file from,
similar to, for example,
<a href="https://services.healthtech.dtu.dk/services/NetMHCIIpan-3.2/data.Linux.tar.gz">https://services.healthtech.dtu.dk/services/NetMHCIIpan-3.2/data.Linux.tar.gz</a>.
Use get_netmhc2pan_data_url to get the default URL.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>netmhc2pan_folder_name</code></td>
<td>
<p>the folder (to be) used by NetMHCIIpan.
From this location, a subfolder for NetMHCIIpan is created.
Use get_default_netmhc2pan_folder to see the location
of the default NetMHCIIpan folder.
Use get_default_netmhc2pan_subfolder to see the location
of the default NetMHCIIpan subfolder.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>netmhc2pan_subfolder</code></td>
<td>
<p>the subfolder (to be) used by NetMHCIIpan.
Use get_default_netmhc2pan_subfolder to see the location
of the default NetMHCIIpan subfolder</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>netmhc2pan_version</code></td>
<td>
<p>the NetMHCIIpan version,
for example <code>3.2</code>.
Use get_default_netmhc2pan_version to get the default
NetMHCIIpan version</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>os</code></td>
<td>
<p>the operating system as obtained by <code>rappdirs::app_dir()$os</code>.
<code>netmhc2pan</code> supports Linux ('<code>unix</code>') only</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>peptide_length</code></td>
<td>
<p>length of a peptide</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>peptides</code></td>
<td>
<p>one or more peptide sequences</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>protein_sequence</code></td>
<td>
<p>a protein sequence, for example <code>FAMILYVW</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>temp_fasta_filename</code></td>
<td>
<p>name for a temporary FASTA file,
which will be deleted automatically</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>temp_local_file</code></td>
<td>
<p>path to the temporary file to store the URL to.
This file will be deleted afterwards.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>temp_xls_filename</code></td>
<td>
<p>name for a temporary <code>xls</code> file,
which will be deleted automatically</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>set to TRUE for more output</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xls_filename</code></td>
<td>
<p>name of an <code>xls</code> file</p>
</td>
</tr>
</table>
<h3>Note</h3>

<p>This is an internal function, so it should be marked with
<code>@noRd</code>. This is not done, as this will disallow all
functions to find the documentation parameters
</p>


<h3>Author(s)</h3>

<p>Rich√®l J.C. Bilderbeek
</p>


</div>