<div class="container">

<table style="width: 100%;"><tr>
<td>NewmanPaired-class</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Class "NewmanPaired"</h2>

<h3>Description</h3>

<p>Represents the reusults of computing the Newman Paired test statistic
on one or more paired samples.
</p>


<h3>Creating Objects</h3>

<p>In practice, users will use the <code>pairedStat</code> function to
construct an object of the <code>NewmanPaired</code> class. Hand
construction is strongly discouraged.
</p>


<h3>Slots</h3>


<dl>
<dt>
<code>pairedMean</code>:</dt>
<dd>
<p>A matrix of size N (number of features) by S (number
of sample pairs). The mean expression of each feature in each paired
sample. Also called "A" in the M-versus-A plots of the microarray era.</p>
</dd>
<dt>
<code>difference</code>:</dt>
<dd>
<p>A matrix of size N (number of features) by S (number
of sample pairs). The difference (perturbed - base) in expression of
each feature in each paired sample. Also called "M" in the M-versus-A
plots of the microarray era.</p>
</dd>
<dt>
<code>smoothSD</code>:</dt>
<dd>
<p>A matrix of size N (number of features) by S (number
of sample pairs). The results of fitting a loess smooth to the
relationship between the <code>PairedMean</code> and the observed estimate
of standard deviation (i.e., <code>abs(difference)/sqrt(2)</code>).</p>
</dd>
<dt>
<code>nuStatistics</code>:</dt>
<dd>
<p>A matrix of size N (number of features) by S (number
of sample pairs). The Newman paired statistics, nu.</p>
</dd>
<dt>
<code>pValues</code>:</dt>
<dd>
<p>A matrix of size N (number of features) by S (number
of sample pairs). Empirical p-values for the Newman statistics.</p>
</dd>
</dl>
<h3>Methods</h3>


<dl>
<dt>x[i,j]</dt>
<dd>
<p>Select a subset of features or sample pairs.</p>
</dd>
<dt>dim(x)</dt>
<dd>
<p>The dimension, N by S, of the object.</p>
</dd>
<dt>plot(x, y, which = NULL, ask = NULL, high = 0.99, low = 0.01,
...)</dt>
<dd>
<p>Plot the results of the analysis of one sample pair.</p>
</dd>
<dt>hist(x, breaks=101, xlab="P-value", ...)</dt>
<dd>
<p>Plot a histogram of the p-values for one sample-pair.</p>
</dd>
</dl>
<h3>Author(s)</h3>

<p>Kevin R. Coombes <a href="mailto:krc@silicovore.com">krc@silicovore.com</a>
</p>


<h3>References</h3>

<p>Abrams ZB, Joglekar A, Gershkowitz GR, Sinicropi-yao S, Asiaee A,
Carbone DP, Coombes KR. Personalized Transcriptomics: Selecting Drugs
Based on Gene Expression Profiles. Preprint.
</p>


<h3>See Also</h3>

<p><code>pairedStat</code>, <code>bankStat</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">showClass("NewmanPaired")
</code></pre>


</div>