<div class="container">

<table style="width: 100%;"><tr>
<td>pairedStat</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Paired Newman Statistic</h2>

<h3>Description</h3>

<p>The Paired Newman Statistic is used for one-to-one comparison of
paired individual samples. Commonly used to find differential expression
between tumor-normal pairs or before-after treatment pairs.
</p>


<h3>Usage</h3>

<pre><code class="language-R">pairedStat(baseData, perturbedData = NULL, pairing = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>baseData</code></td>
<td>
<p>Either a list or a matrix. May contain data for just the
base condition (for example, normal samples or samples before
treatment) or for both the base condition and the perturbed condition
(for example, tumor samples or samples after treatment). See details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>perturbedData</code></td>
<td>
<p>An optional matrix containing data for the "perturbed"
samples. May be NULL if the <code>baseData</code> argument is a list or a
matrix containing all the data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pairing</code></td>
<td>
<p>An optional vector indicating the pairing between base
and perturbed samples. Entries must be integers. Positive integers
indicate perturbed samples and negative integers with the same
absolute value indicate the paired base samples. See details.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>In the simplest case, we have gene expression data on one "base"
sample and one "perturbed" sample, and the goal is to identify genes
whose expression changes between the two states. Our primary
assumption is that the standard deviation (SD) of gene expression
varies as a smooth function of the mean; fitting such a curve allows
us to detect individual genes whose difference is large compared to
the smoothed SD.
</p>
<p>Note that this assumption is most useful on the log-transformed
scale  (<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4120293/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4120293/</a>).
If your data is on a raw scale, then we recommend transforming
it before computing the Newman paired statistic. 
</p>
<p>The input arguments to the <code>pairedStats</code> function are moderately
complicated in order to allow users to choose a convenient method for
supplying data when they have multiple paired samples. The first
posssibility is to have all the base samples in one matrix and all the
perturbed samples in a second matrix. In this case, we assume (without
checking) that the columns in the two matrices correspond to the
paired samples, and that the genes-rows are in the same order.
</p>
<p>The second possibility is to put the data for both the base samples
and the perturbed samples in the same matrix. In this case, the user
must supply a <code>pairing</code> vector to explain how the samples should
be matched. If the column order is ("base1", "perturbed1", "base2",
"perturbed2", ...), then the pairiing vector should be written as
<code>c(-1, 1, -2, 2, -3, 3, ...)</code>.
</p>
<p>The third possibility is to provide the paired samples in a list,
each of whose entries is a matrix with two columns,with the first
column being the base state and the second column being the
perturbed state.
</p>
<p>This flexibility means that there are three equivalent ways to input
the data even if you have only one base sample (with data in the
one-column matrix B) and one perturbed sample (with data in the
one-column matrix P). If we let <code>BP &lt;- cbind(B, P) </code>, then we can
choose (1) <code>pairedStats(B, P)</code>, or (2)
<code>pairedStats(list(BP))</code>, or (3)  <code>pairedStats(BP,
  pairing = c(-1,1))</code>. 
</p>


<h3>Value</h3>

<p>A list containing two marices: the <code>nu.statistics</code> and the <code>p.values</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(GSE6631)
Normal &lt;- GSE6631[, c(1,3)]
Tumor &lt;- GSE6631[, c(2,4)]

### input two separate matrices
ps1 &lt;- pairedStat(Normal, Tumor)
summary(ps1@nu.statistics)
summary(ps1@p.values)

### input one combined matrix and a pairing vector
ps2 &lt;- pairedStat(GSE6631, pairing=c(-1, 1, -2, 2))
summary(ps2@nu.statistics)
summary(ps2@p.values)

### input a list of matrix-pairs
ps3 &lt;- pairedStat(list(One = GSE6631[, 1:2],
                       Two = GSE6631[, 3:4]))
summary(ps3@nu.statistics)
summary(ps3@p.values)
</code></pre>


</div>