<div class="container">

<table style="width: 100%;"><tr>
<td>nodeSub-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Package providing functions to simulate sequences under different DNA evolution models
</h2>

<h3>Description</h3>

<p>Simulate DNA sequences for the node substitution model.
In the node substitution model, substitutions accumulate additionally
during a speciation event, providing a potential mechanistic explanation for
substitution rate variation. This package provides tools to simulate
such a process, simulate a reference process with only substitutions along
the branches, and provides tools to infer phylogenies from alignments. More
information can be found in Janzen (2021) &lt;doi:10.1093/sysbio/syab085&gt;.
</p>
<p>Version History: <br>
Version 1.2.7   -  Removed beta calculation due to apTreeshape removal from
CRAN <br>
Version 1.2.3   -  Removed summary statistic tests for CRAN <br>
Version 1.2.2   -  Changed codedov links in README <br>
Version 1.2.1   -  Expanded depenency on RPANDA <br>
Version 1.2     -  Release on CRAN <br></p>


<h3>Author(s)</h3>

<p>Thijs Janzen
Maintainer: Thijs Janzen &lt;t.janzen@rug.nl&gt;
</p>


<h3>References</h3>

<p>Thijs Janzen, Folmer Bokma, Rampal S Etienne, Nucleotide Substitutions during Speciation may Explain Substitution Rate Variation, Systematic Biology, 2021; syab085
</p>


</div>