<div class="container">

<table style="width: 100%;"><tr>
<td>Genetic regression</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Linear and Multilinear Genetic Regressions </h2>

<h3>Description</h3>

<p>The regression aims at estimating genetic effects from a population in which the genotypes and phenotypes are known. 
</p>


<h3>Usage</h3>

<pre><code class="language-R">linearRegression(phen, gen=NULL, genZ=NULL, 
    reference="noia", max.level=NULL, max.dom=NULL, fast=FALSE)
multilinearRegression(phen, gen=NULL, genZ=NULL, 
    reference="noia", max.level=NULL, max.dom=NULL, fast=FALSE, 
    e.unique=FALSE, start.algo = "linear", start.values=NULL, 
    robust=FALSE, bilinear.steps=1, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>phen</code></td>
<td>
<p> The vector of individual phenotypes measured in the population. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gen</code></td>
<td>
<p> The matrix of individual genotypes in the population, one column per locus. See <code>genNames</code> for the genotype encoding. Not necessary if <code>genZ</code> is provided. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genZ</code></td>
<td>
<p> The matrix of individual genotypic probabilities in the population, 3 columns per locus, corresponding of the probability of each of the 3 genotypes (the sum must be 1). Not necessary if <code>gen</code> is provided. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>reference</code></td>
<td>
<p> The reference point from which the regression is performed. By default, the <code>"noia"</code> reference point is used, since it provides a fairly good orthogonality. Other possibilities are <code>"G2A"</code>, <code>"F2"</code>, <code>"F1"</code>, <code>"Finf"</code>, <code>"UWR"</code>, <code>"P1"</code> and <code>"P2"</code>. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max.level</code></td>
<td>
<p> Maximum level of interactions. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max.dom</code></td>
<td>
<p> Maximum level for dominance effects. Does not have any effect if &gt;= <code>max.level</code>. In the multilinear regression, the maximum level for dominance effects cannot be &gt; 1. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fast</code></td>
<td>
<p> This "fast" algorithm should be used when (i) the number of loci is high (&gt; 8) and (ii) there are uncertainties in the dataset (missing values or Haley-Knott regression). This algorithm computes the regression matrix directly function, i.e. without computing <code>Z</code> nor <code>S</code> matrices.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>e.unique</code></td>
<td>
<p> Whether the multilinear term is the same for all pairs. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>start.algo</code></td>
<td>
<p> Algorithm used to compute the starting values. Can be <code>"linear"</code>, <code>"multilinear"</code>, <code>"subset"</code> or <code>"bilinear"</code>. Ignored if <code>start.values</code> are provided.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>start.values</code></td>
<td>
<p> Vector of starting values. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>robust</code></td>
<td>
<p> Tries sequentially all starting values algorithms.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bilinear.steps</code></td>
<td>
<p> Number of steps. Ignored if <code>start.algo</code> is not <code>"bilinear"</code>. If <code>NULL</code>, the bilinear algorithm is run until (almost) convergence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p> Extra parameters to the non-linear regression function <code>nls</code>, including <code>nls.control</code>. </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>If a <code>gen</code> data set is provided, it will be turned into a <code>genZ</code>. Missing data (unknown genotypes) are considered as loci for which genotypic probabilities are identical to the genotypic frequencies in the population. 
</p>
<p>The algebraic framework is described extensively in Alvarez-Castro &amp; Carlborg 2007. The default reference point (<code>"noia"</code>) provides an orthogonal decomposition of genetic effects in the 1-locus case, whatever the genotypic frequencies. It remains a good approximation of orthogonality in the multi-locus case if linkage disequilibrium is small. Other optional reference points are those of the <code>"G2A"</code> model (Zeng et al. 2005), and the unweighted regression model <code>"UWR"</code> (Cheverud &amp; Routman, 1995). Several key populations can be taken as reference as well: <code>"F2"</code>, <code>"F1"</code>, <code>"Finf"</code> (F infinity), and the two "parental" homozygous populations <code>"P1"</code> and <code>"P2"</code>. 
</p>
<p>The multilinear model for genetic interactions is an alternative way to model epistatic interactions between at least two loci (see Hansen &amp; Wagner 2001). The computation of multilinear estimates requires a non-linear regression step that relies on the <code>nls</code> function. Providing good starting values for the non-linear regression is a key to ensure convergence, and different algorithms are provided, that can be specified by the <code>"start.algo"</code> option. <code>"linear"</code> performs a linear regression and approximates the genetic effects from it, while <code>"multilinear"</code> performs a simpler multilinear regression (without dominance) to initialize the genetic effects. <code>"subset"</code> estimate all genetic effects from a random subset (50%) of the population, and <code>"bilinear"</code> estimate alternatively marginal and epistatic effects. 
</p>


<h3>Value</h3>

<p><code>linearRegression</code> and <code>multilinearRegression</code> return an object of class <code>"noia.linear"</code> or <code>"noia.multilinear"</code>, both having their own <code>print</code> methods: <code>print.noia.linear</code> and <code>print.noia.multilinear</code>. 
</p>


<h3>Author(s)</h3>

<p> Arnaud Le Rouzic </p>


<h3>References</h3>

 
<p>Alvarez-Castro JM, Carlborg O. (2007). A unified model for functional 
and statistical epistasis and its application in quantitative trait 
loci analysis. Genetics 176(2):1151-1167.
</p>
<p>Alvarez-Castro JM, Le Rouzic A, Carlborg O. (2008). How to perform meaningful
estimates of genetic effects. PLoS Genetics 4(5):e1000062.
</p>
<p>Cheverud JM, Routman, EJ. (1995). Epistasis and its contribution to 
genetic variance components. Genetics 139:1455-1461.
</p>
<p>Hansen TF, Wagner G. (2001) Modeling genetic architecture: A multilinear 
theory of gene interactions. Theoretical Population Biology 59:61-86.
</p>
<p>Le Rouzic A, Alvarez-Castro JM. (2008). Estimation of genetic effects and
genotype-phenotype maps. Evolutionary Bioinformatics 4. 
</p>
<p>Zeng ZB, Wang T, Zou W. (2005). Modelling quantitative trait loci and 
interpretation of models. Genetics 169: 1711-1725. 
</p>


<h3>See Also</h3>

 <p><code>geneticEffects</code>, <code>GPmap</code>, <code>varianceDecomposition</code>. </p>


<h3>Examples</h3>

<pre><code class="language-R">set.seed(123456789)

map &lt;- c(0.25, -0.75, -0.75, -0.75, 2.25, 2.25, -0.75, 2.25, 2.25)
pop &lt;- simulatePop(map, N=500, sigmaE=0.2, type="F2")

# Regressions

linear &lt;- linearRegression(phen=pop$phen, gen=cbind(pop$Loc1, pop$Loc2))

multilinear &lt;- multilinearRegression(phen=pop$phen, 
    gen=cbind(pop$Loc1, pop$Loc2))

# Linear effects, associated variances and stderr
linear

# Multilinear effects
multilinear
</code></pre>


</div>