<div class="container">

<table style="width: 100%;"><tr>
<td>GP map analysis</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Noia analysis of genotype-to-phenotype (GP) maps in ideal populations </h2>

<h3>Description</h3>

<p> Functions for doing a NOIA analysis of a GP map for <code class="reqn">L</code> loci in a population where the loci are in complete linkage equilibrium. </p>


<h3>Usage</h3>

<pre><code class="language-R">linearGPmapanalysis(gmap, reference="F2", freqmat=NULL, 
                    max.level=NULL , S_full=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>gmap</code></td>
<td>
<p> Vector of length <code class="reqn">3^L</code> with genotypic values for all possible genotypes in the order defined by <code>genNames</code>. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>reference</code></td>
<td>
<p> The reference population in which the analysis is done. By default, the <code>"F2"</code> population is used. Other possibilities are <code>"noia"</code>, <code>"G2A"</code>, <code>"UWR"</code>. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>freqmat</code></td>
<td>
<p> For <code>reference="G2A"</code>: A vector of length <code class="reqn">L</code> containing allele frequencies such that 
<code>freqmat[i]=frequency(allele 1)</code> for locus <code>i</code>.
</p>
<p>For <code>reference="noia"</code>: A <code class="reqn">(L\times3)</code> matrix of genotype frequencies such that 
<code>freqmat[i,]=[frequency(1) frequency(2) frequency(3)]</code> for locus <code>i</code>. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max.level</code></td>
<td>
<p> Maximum level of interactions. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>S_full</code></td>
<td>
<p> Boolean argument indicating whether to keep full <code>S</code> matrix <code class="reqn">(3^L\times3^L)</code> in memory or alternatively to keep 
<code class="reqn">L</code> single locus <code>S</code> matrices <code class="reqn">(3\times3)</code> and compute single row and columns of the full matrix. </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The algebraic framework is described extensively in Alvarez-Castro &amp; Carlborg 2007. When analysing GP maps in ideal populations
we can work directly with the <code>S</code> matrix and do not have to consider the <code>X</code> and <code>Z</code> matrices used in <code>linearRegression</code>.
When it comes to the <code>S_full</code> argument keeping the multilocus <code>S</code> matrix in memory is generally fastest for computing all <code class="reqn">3^L</code> 
genetic effects. However it does not allow for computing only a subset of the effects and also runs out of memory for <code class="reqn">L&gt;8</code> on a typical desktop machine.
For S_full=NULL in <code>linearGPmapanalysis</code> a full <code>S</code> matrix is used if <code class="reqn">L&lt;=8</code> and max.level=NULL, while <code class="reqn">L</code> single locus <code>S</code> matrices are used otherwise.  
</p>


<h3>Value</h3>

<p><code>linearGPmapanalysis</code> returns an object of class <code>"noia.linear.gpmap"</code> , with its own <code>print</code> method: <code>print.noia.linear.gpmap</code>. 
</p>


<h3>Author(s)</h3>

<p> Arne B. Gjuvsland </p>


<h3>References</h3>

 
<p>Alvarez-Castro JM, Carlborg O. (2007). A unified model for functional 
and statistical epistasis and its application in quantitative trait 
loci analysis. Genetics 176(2):1151-1167.
</p>
<p>Cheverud JM, Routman, EJ. (1995). Epistasis and its contribution to 
genetic variance components. Genetics 139:1455-1461.
</p>
<p>Le Rouzic A, Alvarez-Castro JM. (2008). Estimation of genetic effects and
genotype-phenotype maps. Evolutionary Bioinformatics 4. 
</p>
<p>Zeng ZB, Wang T, Zou W. (2005). Modelling quantitative trait loci and 
interpretation of models. Genetics 169: 1711-1725. 
</p>


<h3>See Also</h3>

 <p><code>varianceDecomposition</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">map &lt;- c(0.25, -0.75, -0.75, -0.75, 2.25, 2.25, -0.75, 2.25, 2.25)

# Genotype-to-phenotype map analysis
linearGP &lt;- linearGPmapanalysis(map, reference="F2")

# Linear effects in ideal F2 population
linearGP

</code></pre>


</div>