<div class="container">

<table style="width: 100%;"><tr>
<td>cast_gtf_to_genes</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Function to extract exon names and positions from a gtf file</h2>

<h3>Description</h3>

<p>This function is used to extract all exons and their positions
in the genome from an input gtf file.
</p>


<h3>Usage</h3>

<pre><code class="language-R">cast_gtf_to_genes(filename, feature = "exon", att_of_interest = "gene_id", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>filename</code></td>
<td>
<p>path to the gtf file</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>feature</code></td>
<td>
<p>the feature type name to filter the feature (3rd) column of the gtf/gff file;
default is exon</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>att_of_interest</code></td>
<td>
<p>the attribute to extract from the last column of the gtf/gff file;
default in gene_id</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>arguments passed on to other methods</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A tibble of the ids, gene names, chromosomes, start and end positions
of each exon found in the gtf file.
</p>


<h3>Examples</h3>

<pre><code class="language-R">fl &lt;- system.file("extdata", "example.gtf.gz", package="Rsamtools", mustWork=TRUE)
cast_gtf_to_genes(fl)
</code></pre>


</div>