<div class="container">

<table style="width: 100%;"><tr>
<td>filter_genes_transcript</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Function to filter the gene table for the transcript approach</h2>

<h3>Description</h3>

<p>This function is used to filter the gene table (usually created with
<code>cast_gtf_to_genes</code>), only keeping genes above the noise thresholds.
It uses as input the gene table (usually containing individual exons),
an expression matrix for each of these and a vector of abundance thresholds.
This function is used internally by <code>remove_noise_from_bams</code> to determine
which genes to retain.
</p>


<h3>Usage</h3>

<pre><code class="language-R">filter_genes_transcript(
  genes,
  expression.matrix,
  noise.thresholds,
  filter.by = c("gene", "exon"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>genes</code></td>
<td>
<p>a tibble of the exons extracted from the gtf file;
(usually the the output of <code>cast_gtf_to_genes</code>)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>expression.matrix</code></td>
<td>
<p>the expression matrix, usually
calculated by <code>calculate_expression_similarity_transcript</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>noise.thresholds</code></td>
<td>
<p>a vector of expression thresholds by sample</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filter.by</code></td>
<td>
<p>Either "gene" (default) or "exon"; if filter.by="gene", a gene
(as determined by its ENSEMBL id) is removed
if and only if all of its exons are below the corresponding noise thresholds;
if filter.by="exon", then each exon is individually removed
if it is below the corresponding noise thresholds.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>arguments passed on to other methods</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns a filtered tibble of exons, with the noise removed.
</p>


<h3>Examples</h3>

<pre><code class="language-R">bams &lt;- rep(system.file("extdata", "ex1.bam", package="Rsamtools", mustWork=TRUE), 2)
genes &lt;- data.frame("id" = 1:2,
                    "gene_id" = c("gene1", "gene2"),
                    "seqid" = c("seq1", "seq2"),
                    "start" = 1,
                    "end" = 1600)
noise.thresholds &lt;- c(0, 1)
expression.summary = calculate_expression_similarity_transcript(
  bams = bams,
  genes = genes,
  mapq.unique = 99
)
filter_genes_transcript(
    genes = genes,
    expression.matrix = expression.summary$expression.matrix,
    noise.thresholds = noise.thresholds,
)

</code></pre>


</div>