<div class="container">

<table style="width: 100%;"><tr>
<td>calculate_expression_profile</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculate the expression profile of a gene</h2>

<h3>Description</h3>

<p>This function calculates the expression profile of an exon in a selection
of BAM files. The expression profile is defined as the number of reads overlapping
with each position of the exon's transcript.
</p>


<h3>Usage</h3>

<pre><code class="language-R">calculate_expression_profile(
  gene,
  bams,
  unique.only = TRUE,
  mapq.unique = c(50, 255),
  slack = 200
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>gene</code></td>
<td>
<p>The exon for which the expression profile is calculated; this should be
a row from the tibble generated by <code>cast_gtf_to_genes</code>; for a manual input,
a tibble with 1 row and named columns (seqid, start, end) would be needed</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bams</code></td>
<td>
<p>a vector of paths to the BAM files from which the profile is extracted</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>unique.only</code></td>
<td>
<p>whether only uniquely mapped reads should contribute to the profile;
default is TRUE</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mapq.unique</code></td>
<td>
<p>The values of the mapping quality field in the BAM file that corresponds
to uniquely mapped reads; by default, values of 50 and 255 are used as these correspond to
the most popular aligners, but an adjustment might be needed</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>slack</code></td>
<td>
<p>slack needs to be &gt;=readLength, adjust for efficiency; the default is 200,
as it is higher than most modern sequencing experiments</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>The function outputs a list: the first element is a matrix of expression profiles.
Rows correspond to positions in the exon transcript and each column corresponds to an input
BAM file. Each read is counted for all the positions with which it overlaps (so a read of
length 100 that completely overlaps with the exon would be counted for all 100 positions).
The second list element is a vector of raw expression of the gene in the different BAM files
</p>


<h3>Examples</h3>

<pre><code class="language-R">bams &lt;- rep(system.file("extdata", "ex1.bam", package="Rsamtools", mustWork=TRUE), 2)
genes &lt;- data.frame("id" = 1:2,
                    "gene_id" = c("gene1", "gene2"),
                    "seqid" = c("seq1", "seq2"),
                    "start" = 1,
                    "end" = 100)
profile &lt;- calculate_expression_profile(
  gene = genes[1,],
  bams = bams,
  mapq.unique = 99
)

ggplot2::ggplot(tibble::tibble(y = profile$profile[,1],
                               x = seq_along(y))) +
ggplot2::geom_bar(ggplot2::aes(x, y), stat = "identity") +
ggplot2::theme_minimal()
</code></pre>


</div>