<div class="container">

<table style="width: 100%;"><tr>
<td>Nonpareil.Curve-class</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>A single Nonpareil curve. This object can be produced by
<code>Nonpareil.curve</code> and supports S4 methods <code>plot</code>, <code>summary</code>,
<code>print</code>, and <code>predict</code>. For additional details, see help for
<code>summary.Nonpareil.Curve</code>.</h2>

<h3>Description</h3>

<p>A single Nonpareil curve. This object can be produced by
<code>Nonpareil.curve</code> and supports S4 methods <code>plot</code>, <code>summary</code>,
<code>print</code>, and <code>predict</code>. For additional details, see help for
<code>summary.Nonpareil.Curve</code>.
</p>


<h3>Slots</h3>


<dl>
<dt><code>file</code></dt>
<dd>
<p>Input .npo file.</p>
</dd>
<dt><code>label</code></dt>
<dd>
<p>Name of the dataset.</p>
</dd>
<dt><code>col</code></dt>
<dd>
<p>Color of the dataset.</p>
</dd>
<dt><code>L</code></dt>
<dd>
<p>Read length.</p>
</dd>
<dt><code>AL</code></dt>
<dd>
<p>Adjusted read length (same as L for alignment).</p>
</dd>
<dt><code>R</code></dt>
<dd>
<p>Number of reads.</p>
</dd>
<dt><code>LR</code></dt>
<dd>
<p>Effective sequencing effort used.</p>
</dd>
<dt><code>overlap</code></dt>
<dd>
<p>Minimum read overlap.</p>
</dd>
<dt><code>ksize</code></dt>
<dd>
<p>K-mer size (for kmer kernel only).</p>
</dd>
<dt><code>log.sample</code></dt>
<dd>
<p>Multiplier of the log-sampling (or zero if linear).</p>
</dd>
<dt><code>kernel</code></dt>
<dd>
<p>Read-comparison kernel.</p>
</dd>
<dt><code>version</code></dt>
<dd>
<p>Nonpareil version used.</p>
</dd>
<dt><code>x.obs</code></dt>
<dd>
<p>Rarefied sequencing effort.</p>
</dd>
<dt><code>x.adj</code></dt>
<dd>
<p>Adjusted rarefied sequencing effort.</p>
</dd>
<dt><code>y.red</code></dt>
<dd>
<p>Rarefied redundancy (observed).</p>
</dd>
<dt><code>y.cov</code></dt>
<dd>
<p>Rarefied coverage (corrected).</p>
</dd>
<dt><code>y.sd</code></dt>
<dd>
<p>Standard deviation of rarefied coverage.</p>
</dd>
<dt><code>y.p25</code></dt>
<dd>
<p>Percentile 25 (1st quartile) of rarefied coverage.</p>
</dd>
<dt><code>y.p50</code></dt>
<dd>
<p>Percentile 50 (median) of rarefied coverage.</p>
</dd>
<dt><code>y.p75</code></dt>
<dd>
<p>Percentile 75 (3rd quartile) of rarefied coverage.</p>
</dd>
<dt><code>kappa</code></dt>
<dd>
<p>Dataset redundancy.</p>
</dd>
<dt><code>C</code></dt>
<dd>
<p>Dataset coverage.</p>
</dd>
<dt><code>consistent</code></dt>
<dd>
<p>Is the data sufficient for accurate estimation?</p>
</dd>
<dt><code>star</code></dt>
<dd>
<p>Coverage considered 'nearly complete'.</p>
</dd>
<dt><code>has.model</code></dt>
<dd>
<p>Was the model successfully estimated?</p>
</dd>
<dt><code>warning</code></dt>
<dd>
<p>Warnings generated on consistency or model fitting.</p>
</dd>
<dt><code>LRstar</code></dt>
<dd>
<p>Projected seq. effort for nearly complete coverage.</p>
</dd>
<dt><code>modelR</code></dt>
<dd>
<p>Pearson's R for the estimated model.</p>
</dd>
<dt><code>diversity</code></dt>
<dd>
<p>Dataset Nd index of sequence diversity.</p>
</dd>
<dt><code>model</code></dt>
<dd>
<p>Fitted sigmoidal model.</p>
</dd>
<dt><code>call</code></dt>
<dd>
<p>Call producing this object.</p>
</dd>
</dl>
</div>