<div class="container">

<table style="width: 100%;"><tr>
<td>evalClustLoss</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>ELoss of a partition point estimate compared to a gold standard</h2>

<h3>Description</h3>

<p>Evaluate the loss of a point estimate of the partition compared to a gold standard according to a given loss function
</p>


<h3>Usage</h3>

<pre><code class="language-R">evalClustLoss(c, gs, lossFn = "F-measure", a = 1, b = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>c</code></td>
<td>
<p>vector of length <code>n</code> containing the estimated partition
of the  <code>n</code> observations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gs</code></td>
<td>
<p>vector of length <code>n</code> containing the  gold standard
partition of the  <code>n</code> observations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lossFn</code></td>
<td>
<p>character string specifying the loss function to be used.
Either "F-measure" or "Binder" (see Details). Default is "F-measure".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>a</code></td>
<td>
<p>only relevant if <code>lossFn</code> is "Binder". Penalty for wrong
co-clustering in <code>c</code> compared to <code>gs</code>. Defaults is 1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>b</code></td>
<td>
<p>only relevant if <code>lossFn</code> is "Binder". Penalty for missed
co-clustering in <code>c</code> compared to <code>gs</code>. Defaults is 1.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The cost of a point estimate partition is calculated using either a pairwise
coincidence loss function (Binder), or 1-Fmeasure (F-measure).
</p>


<h3>Value</h3>

<p>the cost of the point estimate <code>c</code> in regard of the
gold standard <code>gs</code> for a given loss function.
</p>


<h3>Author(s)</h3>

<p>Boris Hejblum
</p>


<h3>References</h3>

<p>J.W. Lau &amp; P.J. Green. Bayesian Model-Based Clustering
Procedures, Journal of Computational and Graphical Statistics,
16(3): 526-558, 2007.
</p>
<p>D. B. Dahl. Model-Based Clustering for Expression Data via a
Dirichlet Process Mixture Model, in Bayesian Inference for
Gene Expression and Proteomics, K.-A. Do, P. Muller, M. Vannucci
(Eds.), Cambridge University Press, 2006.
</p>


<h3>See Also</h3>

<p><code>similarityMat</code>, <code>cluster_est_binder</code>
</p>


</div>