<div class="container">

<table style="width: 100%;"><tr>
<td>genomeplot</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Creation of Genome Graph files</h2>

<h3>Description</h3>

<p>The <code>genomeplot</code> function generates two formatted files, one
containing “chromosome base” formatted genome data and the other
containing marker-specific results with dbSNP SNP IDs for displaying  
genome-wide data sets in the UCSC genome browser. </p>


<h3>Usage</h3>

<pre><code class="language-R">genomeplot(gg.data)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>gg.data</code></td>
<td>
<p>a file containing chromosome, marker, physical position and scores. </p>
</td>
</tr></table>
<h3>Details</h3>

<p>gg.data example:
</p>
<pre>
Chromosome   Marker   Position   TRAIT_ALL   
         5   M1       0.000      0.670
         5   -        2.500      0.640
         5   M3       5.000      0.590
         5   -        6.500      0.600
         5   M2       8.000      0.610                   
         8   M4       0.000      0.670
         8   -        2.500      0.640
         8   M6       5.000      0.590
         8   -        6.500      0.600
         8   M5       8.000      0.610 
</pre>
<p>Two files are created, “GG.positons.all” for the
“chromosome base” format, and “GG.markers.all” for the
marker-names based format. When there are multiple scores in
<code>gg.data</code> file, this results in matched pairs of files, one for
each score, labelled with the score names, e.g. GG.positions.score1.all,
and GG.markers.score1.all, GG.positions.score2.all and
GG.markers.score2.all, and so on.
</p>


<h3>Value</h3>

<p>TRUE or FALSE depending on whether runs successfully. </p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: genomeplot("GG.data.all")
</code></pre>


</div>