<div class="container">

<table style="width: 100%;"><tr>
<td>plot_psbulk</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot a pseudobulk HMM profile</h2>

<h3>Description</h3>

<p>Plot a pseudobulk HMM profile
</p>


<h3>Usage</h3>

<pre><code class="language-R">plot_psbulk(
  bulk,
  use_pos = TRUE,
  allele_only = FALSE,
  min_LLR = 5,
  min_depth = 8,
  exp_limit = 2,
  phi_mle = TRUE,
  theta_roll = FALSE,
  dot_size = 0.8,
  dot_alpha = 0.5,
  legend = TRUE,
  exclude_gap = TRUE,
  genome = "hg38",
  text_size = 10,
  raster = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>bulk</code></td>
<td>
<p>dataframe Pseudobulk profile</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>use_pos</code></td>
<td>
<p>logical Use marker position instead of index as x coordinate</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>allele_only</code></td>
<td>
<p>logical Only plot alleles</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min_LLR</code></td>
<td>
<p>numeric LLR threshold for event filtering</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min_depth</code></td>
<td>
<p>numeric Minimum coverage depth for a SNP to be plotted</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exp_limit</code></td>
<td>
<p>numeric Expression logFC axis limit</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>phi_mle</code></td>
<td>
<p>logical Whether to plot estimates of segmental expression fold change</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>theta_roll</code></td>
<td>
<p>logical Whether to plot rolling estimates of allele imbalance</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dot_size</code></td>
<td>
<p>numeric Size of marker dots</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dot_alpha</code></td>
<td>
<p>numeric Transparency of the marker dots</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>legend</code></td>
<td>
<p>logical Whether to show legend</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exclude_gap</code></td>
<td>
<p>logical Whether to mark gap regions and centromeres</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genome</code></td>
<td>
<p>character Genome build, either 'hg38' or 'hg19'</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>text_size</code></td>
<td>
<p>numeric Size of text in the plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>raster</code></td>
<td>
<p>logical Whether to raster images</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>ggplot Plot of pseudobulk HMM profile
</p>


<h3>Examples</h3>

<pre><code class="language-R">p = plot_psbulk(bulk_example)
</code></pre>


</div>