<div class="container">

<table style="width: 100%;"><tr>
<td>run_numbat</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Run workflow to decompose tumor subclones</h2>

<h3>Description</h3>

<p>Run workflow to decompose tumor subclones
</p>


<h3>Usage</h3>

<pre><code class="language-R">run_numbat(
  count_mat,
  lambdas_ref,
  df_allele,
  genome = "hg38",
  out_dir = tempdir(),
  max_iter = 2,
  max_nni = 100,
  t = 1e-05,
  gamma = 20,
  min_LLR = 5,
  alpha = 1e-04,
  eps = 1e-05,
  max_entropy = 0.5,
  init_k = 3,
  min_cells = 50,
  tau = 0.3,
  nu = 1,
  max_cost = ncol(count_mat) * tau,
  n_cut = 0,
  min_depth = 0,
  common_diploid = TRUE,
  min_overlap = 0.45,
  ncores = 1,
  ncores_nni = ncores,
  random_init = FALSE,
  segs_loh = NULL,
  call_clonal_loh = FALSE,
  verbose = TRUE,
  diploid_chroms = NULL,
  segs_consensus_fix = NULL,
  use_loh = NULL,
  min_genes = 10,
  skip_nj = FALSE,
  multi_allelic = TRUE,
  p_multi = 1 - alpha,
  plot = TRUE,
  check_convergence = FALSE,
  exclude_neu = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>count_mat</code></td>
<td>
<p>dgCMatrix Raw count matrices where rownames are genes and column names are cells</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lambdas_ref</code></td>
<td>
<p>matrix Either a named vector with gene names as names and normalized expression as values, or a matrix where rownames are genes and columns are pseudobulk names</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>df_allele</code></td>
<td>
<p>dataframe Allele counts per cell, produced by preprocess_allele</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genome</code></td>
<td>
<p>character Genome version (hg38, hg19, or mm10)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>out_dir</code></td>
<td>
<p>string Output directory</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max_iter</code></td>
<td>
<p>integer Maximum number of iterations to run the phyologeny optimization</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max_nni</code></td>
<td>
<p>integer Maximum number of iterations to run NNI in the ML phylogeny inference</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>t</code></td>
<td>
<p>numeric Transition probability</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gamma</code></td>
<td>
<p>numeric Dispersion parameter for the Beta-Binomial allele model</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min_LLR</code></td>
<td>
<p>numeric Minimum LLR to filter CNVs</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p>numeric P value cutoff for diploid finding</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>eps</code></td>
<td>
<p>numeric Convergence threshold for ML tree search</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max_entropy</code></td>
<td>
<p>numeric Entropy threshold to filter CNVs</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>init_k</code></td>
<td>
<p>integer Number of clusters in the initial clustering</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min_cells</code></td>
<td>
<p>integer Minimum number of cells to run HMM on</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tau</code></td>
<td>
<p>numeric Factor to determine max_cost as a function of the number of cells (0-1)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nu</code></td>
<td>
<p>numeric Phase switch rate</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max_cost</code></td>
<td>
<p>numeric Likelihood threshold to collapse internal branches</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n_cut</code></td>
<td>
<p>integer Number of cuts on the phylogeny to define subclones</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min_depth</code></td>
<td>
<p>integer Minimum allele depth</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>common_diploid</code></td>
<td>
<p>logical Whether to find common diploid regions in a group of peusdobulks</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min_overlap</code></td>
<td>
<p>numeric Minimum CNV overlap threshold</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncores</code></td>
<td>
<p>integer Number of threads to use</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncores_nni</code></td>
<td>
<p>integer Number of threads to use for NNI</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>random_init</code></td>
<td>
<p>logical Whether to initiate phylogney using a random tree (internal use only)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>segs_loh</code></td>
<td>
<p>dataframe Segments of clonal LOH to be excluded</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>call_clonal_loh</code></td>
<td>
<p>logical Whether to call segments with clonal LOH</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>logical Verbosity</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>diploid_chroms</code></td>
<td>
<p>vector Known diploid chromosomes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>segs_consensus_fix</code></td>
<td>
<p>dataframe Pre-determined segmentation of consensus CNVs</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>use_loh</code></td>
<td>
<p>logical Whether to include LOH regions in the expression baseline</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min_genes</code></td>
<td>
<p>integer Minimum number of genes to call a segment</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>skip_nj</code></td>
<td>
<p>logical Whether to skip NJ tree construction and only use UPGMA</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>multi_allelic</code></td>
<td>
<p>logical Whether to call multi-allelic CNVs</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p_multi</code></td>
<td>
<p>numeric P value cutoff for calling multi-allelic CNVs</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot</code></td>
<td>
<p>logical Whether to plot results</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>check_convergence</code></td>
<td>
<p>logical Whether to terminate iterations based on consensus CNV convergence</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exclude_neu</code></td>
<td>
<p>logical Whether to exclude neutral segments from CNV retesting (internal use only)</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a status code
</p>


</div>