<div class="container">

<table style="width: 100%;"><tr>
<td>plot_exp_roll</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot single-cell smoothed expression magnitude heatmap</h2>

<h3>Description</h3>

<p>Plot single-cell smoothed expression magnitude heatmap
</p>


<h3>Usage</h3>

<pre><code class="language-R">plot_exp_roll(
  gexp_roll_wide,
  hc,
  k,
  gtf,
  lim = 0.8,
  n_sample = 300,
  reverse = TRUE,
  plot_tree = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>gexp_roll_wide</code></td>
<td>
<p>matrix Cell x gene smoothed expression magnitudes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hc</code></td>
<td>
<p>hclust Hierarchical clustring result</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>k</code></td>
<td>
<p>integer Number of clusters</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gtf</code></td>
<td>
<p>dataframe Transcript GTF</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lim</code></td>
<td>
<p>numeric Limit for expression magnitudes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n_sample</code></td>
<td>
<p>integer Number of cells to subsample</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>reverse</code></td>
<td>
<p>logical Whether to reverse the cell order</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot_tree</code></td>
<td>
<p>logical Whether to plot the dendrogram</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>ggplot A single-cell heatmap of window-smoothed expression CNV signals
</p>


<h3>Examples</h3>

<pre><code class="language-R">p = plot_exp_roll(gexp_roll_example, gtf = gtf_hg38, hc = hc_example, k = 3)
</code></pre>


</div>