<div class="container">

<table style="width: 100%;"><tr>
<td>plot_phylo_heatmap</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot single-cell CNV calls along with the clonal phylogeny</h2>

<h3>Description</h3>

<p>Plot single-cell CNV calls along with the clonal phylogeny
</p>


<h3>Usage</h3>

<pre><code class="language-R">plot_phylo_heatmap(
  gtree,
  joint_post,
  segs_consensus,
  clone_post = NULL,
  p_min = 0.9,
  annot = NULL,
  pal_annot = NULL,
  annot_title = "Annotation",
  annot_scale = NULL,
  clone_dict = NULL,
  clone_bar = TRUE,
  clone_stack = TRUE,
  pal_clone = NULL,
  clone_title = "Genotype",
  clone_legend = TRUE,
  line_width = 0.1,
  tree_height = 1,
  branch_width = 0.2,
  tip_length = 0.2,
  annot_bar_width = 0.25,
  clone_bar_width = 0.25,
  bar_label_size = 7,
  tvn_line = TRUE,
  clone_line = FALSE,
  exclude_gap = FALSE,
  root_edge = TRUE,
  raster = FALSE,
  show_phylo = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>gtree</code></td>
<td>
<p>tbl_graph The single-cell phylogeny</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>joint_post</code></td>
<td>
<p>dataframe Joint single cell CNV posteriors</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>segs_consensus</code></td>
<td>
<p>datatframe Consensus segment dataframe</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>clone_post</code></td>
<td>
<p>dataframe Clone assignment posteriors</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p_min</code></td>
<td>
<p>numeric Probability threshold to display CNV calls</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>annot</code></td>
<td>
<p>dataframe Cell annotations, dataframe with 'cell' and additional annotation columns</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pal_annot</code></td>
<td>
<p>named vector Colors for cell annotations</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>annot_title</code></td>
<td>
<p>character Legend title for the annotation bar</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>annot_scale</code></td>
<td>
<p>ggplot scale Color scale for the annotation bar</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>clone_dict</code></td>
<td>
<p>named vector Clone annotations, mapping from cell name to clones</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>clone_bar</code></td>
<td>
<p>logical Whether to display clone bar plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>clone_stack</code></td>
<td>
<p>character Whether to plot clone assignment probabilities as stacked bar</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pal_clone</code></td>
<td>
<p>named vector Clone colors</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>clone_title</code></td>
<td>
<p>character Legend title for the clone bar</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>clone_legend</code></td>
<td>
<p>logical Whether to display the clone legend</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>line_width</code></td>
<td>
<p>numeric Line width for CNV heatmap</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tree_height</code></td>
<td>
<p>numeric Relative height of the phylogeny plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>branch_width</code></td>
<td>
<p>numeric Line width in the phylogeny</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tip_length</code></td>
<td>
<p>numeric Length of tips in the phylogeny</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>annot_bar_width</code></td>
<td>
<p>numeric Width of annotation bar</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>clone_bar_width</code></td>
<td>
<p>numeric Width of clone genotype bar</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bar_label_size</code></td>
<td>
<p>numeric Size of sidebar text labels</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tvn_line</code></td>
<td>
<p>logical Whether to draw line separating tumor and normal cells</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>clone_line</code></td>
<td>
<p>logical Whether to display borders for clones in the heatmap</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exclude_gap</code></td>
<td>
<p>logical Whether to mark gap regions</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>root_edge</code></td>
<td>
<p>logical Whether to plot root edge</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>raster</code></td>
<td>
<p>logical Whether to raster images</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>show_phylo</code></td>
<td>
<p>logical Whether to display phylogeny on y axis</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>ggplot panel
</p>


<h3>Examples</h3>

<pre><code class="language-R">p = plot_phylo_heatmap(
    gtree = phylogeny_example,
    joint_post = joint_post_example,
    segs_consensus = segs_example)
</code></pre>


</div>