<div class="container">

<table style="width: 100%;"><tr>
<td>analyze_bulk</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Call CNVs in a pseudobulk profile using the Numbat joint HMM</h2>

<h3>Description</h3>

<p>Call CNVs in a pseudobulk profile using the Numbat joint HMM
</p>


<h3>Usage</h3>

<pre><code class="language-R">analyze_bulk(
  bulk,
  t = 1e-05,
  gamma = 20,
  theta_min = 0.08,
  logphi_min = 0.25,
  nu = 1,
  min_genes = 10,
  exp_only = FALSE,
  allele_only = FALSE,
  bal_cnv = TRUE,
  retest = TRUE,
  find_diploid = TRUE,
  diploid_chroms = NULL,
  classify_allele = FALSE,
  run_hmm = TRUE,
  prior = NULL,
  exclude_neu = TRUE,
  phasing = TRUE,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>bulk</code></td>
<td>
<p>dataframe Pesudobulk profile</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>t</code></td>
<td>
<p>numeric Transition probability</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gamma</code></td>
<td>
<p>numeric Dispersion parameter for the Beta-Binomial allele model</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>theta_min</code></td>
<td>
<p>numeric Minimum imbalance threshold</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>logphi_min</code></td>
<td>
<p>numeric Minimum log expression deviation threshold</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nu</code></td>
<td>
<p>numeric Phase switch rate</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min_genes</code></td>
<td>
<p>integer Minimum number of genes to call an event</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exp_only</code></td>
<td>
<p>logical Whether to run expression-only HMM</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>allele_only</code></td>
<td>
<p>logical Whether to run allele-only HMM</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bal_cnv</code></td>
<td>
<p>logical Whether to call balanced amplifications/deletions</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>retest</code></td>
<td>
<p>logical Whether to retest CNVs after Viterbi decoding</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>find_diploid</code></td>
<td>
<p>logical Whether to run diploid region identification routine</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>diploid_chroms</code></td>
<td>
<p>character vector User-given chromosomes that are known to be in diploid state</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>classify_allele</code></td>
<td>
<p>logical Whether to only classify allele (internal use only)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>run_hmm</code></td>
<td>
<p>logical Whether to run HMM (internal use only)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prior</code></td>
<td>
<p>numeric vector Prior probabilities of states (internal use only)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exclude_neu</code></td>
<td>
<p>logical Whether to exclude neutral segments from retesting (internal use only)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>phasing</code></td>
<td>
<p>logical Whether to use phasing information (internal use only)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>logical Verbosity</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a pseudobulk profile dataframe with called CNV information
</p>


<h3>Examples</h3>

<pre><code class="language-R">bulk_analyzed = analyze_bulk(bulk_example, t = 1e-5, find_diploid = FALSE, retest = FALSE)
</code></pre>


</div>