<div class="container">

<table style="width: 100%;"><tr>
<td>Numbat</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Numbat R6 class</h2>

<h3>Description</h3>

<p>Used to allow users to plot results
</p>


<h3>Value</h3>

<p>a new 'Numbat' object
</p>


<h3>Public fields</h3>

<div class="r6-fields">

<dl>
<dt><code>label</code></dt>
<dd>
<p>character Sample name</p>
</dd>
<dt><code>gtf</code></dt>
<dd>
<p>dataframe Transcript annotation</p>
</dd>
<dt><code>joint_post</code></dt>
<dd>
<p>dataframe Joint posterior</p>
</dd>
<dt><code>exp_post</code></dt>
<dd>
<p>dataframe Expression posterior</p>
</dd>
<dt><code>allele_post</code></dt>
<dd>
<p>dataframe Allele posetrior</p>
</dd>
<dt><code>bulk_subtrees</code></dt>
<dd>
<p>dataframe Bulk profiles of lineage subtrees</p>
</dd>
<dt><code>bulk_clones</code></dt>
<dd>
<p>dataframe Bulk profiles of clones</p>
</dd>
<dt><code>segs_consensus</code></dt>
<dd>
<p>dataframe Consensus segments</p>
</dd>
<dt><code>tree_post</code></dt>
<dd>
<p>list Tree posterior</p>
</dd>
<dt><code>mut_graph</code></dt>
<dd>
<p>igraph Mutation history graph</p>
</dd>
<dt><code>gtree</code></dt>
<dd>
<p>tbl_graph Single-cell phylogeny</p>
</dd>
<dt><code>clone_post</code></dt>
<dd>
<p>dataframe Clone posteriors</p>
</dd>
<dt><code>gexp_roll_wide</code></dt>
<dd>
<p>matrix Smoothed expression of single cells</p>
</dd>
<dt><code>P</code></dt>
<dd>
<p>matrix Genotype probability matrix</p>
</dd>
<dt><code>treeML</code></dt>
<dd>
<p>matrix Maximum likelihood tree as phylo object</p>
</dd>
<dt><code>hc</code></dt>
<dd>
<p>hclust Initial hierarchical clustering</p>
</dd>
</dl>
</div>


<h3>Methods</h3>



<h4>Public methods</h4>


<ul>
<li> <p><a href="#method-Numbat-new"><code>Numbat$new()</code></a>
</p>
</li>
<li> <p><a href="#method-Numbat-plot_phylo_heatmap"><code>Numbat$plot_phylo_heatmap()</code></a>
</p>
</li>
<li> <p><a href="#method-Numbat-plot_exp_roll"><code>Numbat$plot_exp_roll()</code></a>
</p>
</li>
<li> <p><a href="#method-Numbat-plot_mut_history"><code>Numbat$plot_mut_history()</code></a>
</p>
</li>
<li> <p><a href="#method-Numbat-plot_sc_tree"><code>Numbat$plot_sc_tree()</code></a>
</p>
</li>
<li> <p><a href="#method-Numbat-plot_consensus"><code>Numbat$plot_consensus()</code></a>
</p>
</li>
<li> <p><a href="#method-Numbat-plot_clone_profile"><code>Numbat$plot_clone_profile()</code></a>
</p>
</li>
<li> <p><a href="#method-Numbat-cutree"><code>Numbat$cutree()</code></a>
</p>
</li>
<li> <p><a href="#method-Numbat-clone"><code>Numbat$clone()</code></a>
</p>
</li>
</ul>
<hr>
<a id="method-Numbat-new"></a>



<h4>Method <code>new()</code>
</h4>

<p>initialize Numbat class
</p>


<h5>Usage</h5>

<div class="r"><pre>Numbat$new(out_dir, i = 2, gtf = gtf_hg38, verbose = TRUE)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>out_dir</code></dt>
<dd>
<p>character string Output directory</p>
</dd>
<dt><code>i</code></dt>
<dd>
<p>integer Get results from which iteration (default=2)</p>
</dd>
<dt><code>gtf</code></dt>
<dd>
<p>dataframe Transcript gtf (default=gtf_hg38)</p>
</dd>
<dt><code>verbose</code></dt>
<dd>
<p>logical Whether to output verbose results (default=TRUE)</p>
</dd>
</dl>
</div>



<h5>Returns</h5>

<p>a new 'Numbat' object
</p>


<hr>
<a id="method-Numbat-plot_phylo_heatmap"></a>



<h4>Method <code>plot_phylo_heatmap()</code>
</h4>

<p>Plot the single-cell CNV calls in a heatmap and the corresponding phylogeny
</p>


<h5>Usage</h5>

<div class="r"><pre>Numbat$plot_phylo_heatmap(...)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>...</code></dt>
<dd>
<p>additional parameters passed to plot_phylo_heatmap()</p>
</dd>
</dl>
</div>


<hr>
<a id="method-Numbat-plot_exp_roll"></a>



<h4>Method <code>plot_exp_roll()</code>
</h4>

<p>Plot window-smoothed expression profiles
</p>


<h5>Usage</h5>

<div class="r"><pre>Numbat$plot_exp_roll(k = 3, n_sample = 300, ...)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>k</code></dt>
<dd>
<p>integer Number of clusters</p>
</dd>
<dt><code>n_sample</code></dt>
<dd>
<p>integer Number of cells to subsample</p>
</dd>
<dt><code>...</code></dt>
<dd>
<p>additional parameters passed to plot_exp_roll()</p>
</dd>
</dl>
</div>


<hr>
<a id="method-Numbat-plot_mut_history"></a>



<h4>Method <code>plot_mut_history()</code>
</h4>

<p>Plot the mutation history of the tumor
</p>


<h5>Usage</h5>

<div class="r"><pre>Numbat$plot_mut_history(...)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>...</code></dt>
<dd>
<p>additional parameters passed to plot_mut_history()</p>
</dd>
</dl>
</div>


<hr>
<a id="method-Numbat-plot_sc_tree"></a>



<h4>Method <code>plot_sc_tree()</code>
</h4>

<p>Plot the single cell phylogeny
</p>


<h5>Usage</h5>

<div class="r"><pre>Numbat$plot_sc_tree(...)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>...</code></dt>
<dd>
<p>additional parameters passed to plot_sc_tree()</p>
</dd>
</dl>
</div>


<hr>
<a id="method-Numbat-plot_consensus"></a>



<h4>Method <code>plot_consensus()</code>
</h4>

<p>Plot consensus segments
</p>


<h5>Usage</h5>

<div class="r"><pre>Numbat$plot_consensus(...)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>...</code></dt>
<dd>
<p>additional parameters passed to plot_sc_tree()</p>
</dd>
</dl>
</div>


<hr>
<a id="method-Numbat-plot_clone_profile"></a>



<h4>Method <code>plot_clone_profile()</code>
</h4>

<p>Plot clone cnv profiles
</p>


<h5>Usage</h5>

<div class="r"><pre>Numbat$plot_clone_profile(...)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>...</code></dt>
<dd>
<p>additional parameters passed to plot_clone_profile()</p>
</dd>
</dl>
</div>


<hr>
<a id="method-Numbat-cutree"></a>



<h4>Method <code>cutree()</code>
</h4>

<p>Re-define subclones on the phylogeny.
</p>


<h5>Usage</h5>

<div class="r"><pre>Numbat$cutree(max_cost = 0, n_cut = 0)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>max_cost</code></dt>
<dd>
<p>numeric Likelihood threshold to collapse internal branches</p>
</dd>
<dt><code>n_cut</code></dt>
<dd>
<p>integer Number of cuts on the phylogeny to define subclones</p>
</dd>
</dl>
</div>


<hr>
<a id="method-Numbat-clone"></a>



<h4>Method <code>clone()</code>
</h4>

<p>The objects of this class are cloneable with this method.
</p>


<h5>Usage</h5>

<div class="r"><pre>Numbat$clone(deep = FALSE)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>deep</code></dt>
<dd>
<p>Whether to make a deep clone.</p>
</dd>
</dl>
</div>




</div>