<div class="container">

<table style="width: 100%;"><tr>
<td>get_gtree</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get a tidygraph tree with simplified mutational history.</h2>

<h3>Description</h3>

<p>Specify either max_cost or n_cut. 
max_cost works similarly as h and n_cut works similarly as k in stats::cutree.
The top-level normal diploid clone is always included.
</p>


<h3>Usage</h3>

<pre><code class="language-R">get_gtree(tree, P, n_cut = 0, max_cost = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>tree</code></td>
<td>
<p>phylo Single-cell phylogenetic tree</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>P</code></td>
<td>
<p>matrix Genotype probability matrix</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n_cut</code></td>
<td>
<p>integer Number of cuts on the phylogeny to define subclones</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max_cost</code></td>
<td>
<p>numeric Likelihood threshold to collapse internal branches</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>tbl_graph Phylogeny annotated with branch lengths and mutation events
</p>


</div>