<div class="container">

<table style="width: 100%;"><tr>
<td>get_bulk</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Aggregate single-cell data into combined bulk expression and allele profile</h2>

<h3>Description</h3>

<p>Aggregate single-cell data into combined bulk expression and allele profile
</p>


<h3>Usage</h3>

<pre><code class="language-R">get_bulk(
  count_mat,
  lambdas_ref,
  df_allele,
  gtf,
  subset = NULL,
  min_depth = 0,
  nu = 1,
  segs_loh = NULL,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>count_mat</code></td>
<td>
<p>dgCMatrix Gene expression counts</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lambdas_ref</code></td>
<td>
<p>matrix Reference expression profiles</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>df_allele</code></td>
<td>
<p>dataframe Single-cell allele counts</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gtf</code></td>
<td>
<p>dataframe Transcript gtf</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>subset</code></td>
<td>
<p>vector Subset of cells to aggregate</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min_depth</code></td>
<td>
<p>integer Minimum coverage to filter SNPs</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nu</code></td>
<td>
<p>numeric Phase switch rate</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>segs_loh</code></td>
<td>
<p>dataframe Segments with clonal LOH to be excluded</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>logical Verbosity</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>dataframe Pseudobulk gene expression and allele profile
</p>


<h3>Examples</h3>

<pre><code class="language-R">bulk_example = get_bulk(
    count_mat = count_mat_example,
    lambdas_ref = ref_hca,
    df_allele = df_allele_example,
    gtf = gtf_hg38)
</code></pre>


</div>